<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Results Output for Question 3</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">esoph-micro-cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Results Output for Question 3</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-11
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>esoph-micro-cancer-workflow/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200916code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200916)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200916code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200916)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomGreathouseLabesophmicrocancerworkflowtree285a2fb543b59efbec5b23efbabbad8df5bba922targetblank285a2fba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/tree/285a2fb543b59efbec5b23efbabbad8df5bba922" target="_blank">285a2fb</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomGreathouseLabesophmicrocancerworkflowtree285a2fb543b59efbec5b23efbabbad8df5bba922targetblank285a2fba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/tree/285a2fb543b59efbec5b23efbabbad8df5bba922" target="_blank">285a2fb</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/

Untracked files:
    Untracked:  analysis/heatmaps-dendrograms-species-level.Rmd
    Untracked:  analysis/ordered-logit-model.Rmd
    Untracked:  output/nci-species-lvl-heatmap-01-2021-02-11.pdf
    Untracked:  output/nci-species-lvl-heatmap-01-2021-02-11.png
    Untracked:  output/nci-species-lvl-heatmap-05-2021-02-11.pdf
    Untracked:  output/nci-species-lvl-heatmap-05-2021-02-11.png
    Untracked:  output/nci-specific-otus-heatmap-2021-02-11.pdf
    Untracked:  output/nci-specific-otus-heatmap-2021-02-11.png
    Untracked:  output/tcga-rna-species-lvl-heatmap-001-2021-02-11.pdf
    Untracked:  output/tcga-rna-species-lvl-heatmap-001-2021-02-11.png
    Untracked:  output/tcga-rna-species-lvl-heatmap-01-2021-02-11.pdf
    Untracked:  output/tcga-rna-species-lvl-heatmap-01-2021-02-11.png
    Untracked:  output/tcga-rna-species-lvl-heatmap-05-2021-02-11.pdf
    Untracked:  output/tcga-rna-species-lvl-heatmap-05-2021-02-11.png
    Untracked:  output/tcga-rna-specific-otus-heatmap-2021-02-11.pdf
    Untracked:  output/tcga-rna-specific-otus-heatmap-2021-02-11.png
    Untracked:  output/tcga-wgs-species-lvl-heatmap-001-2021-02-11.pdf
    Untracked:  output/tcga-wgs-species-lvl-heatmap-001-2021-02-11.png
    Untracked:  output/tcga-wgs-species-lvl-heatmap-01-2021-02-11.pdf
    Untracked:  output/tcga-wgs-species-lvl-heatmap-01-2021-02-11.png
    Untracked:  output/tcga-wgs-species-lvl-heatmap-05-2021-02-11.pdf
    Untracked:  output/tcga-wgs-species-lvl-heatmap-05-2021-02-11.png
    Untracked:  output/tcga-wgs-specific-otus-heatmap-2021-02-11.pdf
    Untracked:  output/tcga-wgs-specific-otus-heatmap-2021-02-11.png

Unstaged changes:
    Modified:   analysis/index.Rmd
    Modified:   code/load_packages.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="question-3" class="section level1">
<h1>Question 3</h1>
<pre><code>Q3: Is fuso associated with tumor stage (pTNM) in either data set? Does X bacteria predict stage? Multivariable w/ age, sex, BMI, history of Barrett&#39;s</code></pre>
<p>Add to this analysis:</p>
<ul>
<li>Fusobacterium nucleatum</li>
<li>Streptococcus sanguinis</li>
<li>Campylobacter concisus</li>
<li>Prevotella spp.</li>
</ul>
<p>TCGA drop “not reported” from tumor stage.</p>
</div>
<div id="nci-16s-data" class="section level1">
<h1>NCI 16s data</h1>
<div id="double-checking-data" class="section level2">
<h2>Double Checking Data</h2>
<pre class="r"><code># in long format
table(dat.16s$tumor.stage)</code></pre>
<pre><code>
    0     1     I    II   III    IV 
11088   264  6336 13464  6072  2640 </code></pre>
<pre class="r"><code># by subject
dat &lt;- dat.16s %&gt;% filter(OTU == &quot;Fusobacterium_nucleatum&quot;)
table(dat$tumor.stage)</code></pre>
<pre><code>
  0   1   I  II III  IV 
 42   1  24  51  23  10 </code></pre>
<pre class="r"><code>sum(table(dat$tumor.stage)) # sample size met</code></pre>
<pre><code>[1] 151</code></pre>
<pre class="r"><code>mean.dat &lt;- dat.16s.s %&gt;%
  group_by(tumor.stage, OTU) %&gt;%
  summarize(M = mean(Abundance))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;tumor.stage&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>ggplot(dat.16s.s, aes(x=tumor.stage, y=Abundance))+
  geom_violin()+
  geom_jitter(alpha=0.25,width = 0.25)+
  geom_point(data=mean.dat, aes(x=tumor.stage, y = M), size=2, alpha =0.9, color=&quot;red&quot;)+
  labs(x=&quot;Tumor Stage&quot;, 
       title=&quot;Distribution of abundance across tumor stage&quot;,
       subtitle=&quot;Red dot is average abundnace&quot;)+
  scale_y_continuous(trans=&quot;pseudo_log&quot;)+
  #  breaks=c(0, 10, 100, 200, 300, 400, 500),
  #  limits = c(0,500),
  #  
  facet_wrap(.~OTU, nrow=1, scales=&quot;free&quot;)+
  theme_classic()</code></pre>
<p><img src="figure/ordered-logit-model.Rmd/nci-fig1-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
<p>Stage “1” has only 1 unique sample and will be dropped from subsequent analyses. And remove NA values.</p>
<pre class="r"><code>dat.16s.s &lt;- dat.16s.s %&gt;%
  filter(tumor.stage != &quot;1&quot;)%&gt;%
  mutate(tumor.stage = droplevels(tumor.stage, exclude=c(&quot;1&quot;,NA)))</code></pre>
</div>
<div id="ordered-logistic-regression" class="section level2">
<h2>Ordered Logistic Regression</h2>
<div id="model-1-ts-otu" class="section level3">
<h3>Model 1: TS ~ OTU</h3>
<pre class="r"><code>## fit ordered logit model and store results &#39;m&#39;
fit &lt;- MASS::polr(tumor.stage ~ OTU, data = dat.16s.s, Hess=TRUE)

## view a summary of the model
summary(fit)</code></pre>
<pre><code>Call:
MASS::polr(formula = tumor.stage ~ OTU, data = dat.16s.s, Hess = TRUE)

Coefficients:
                                 Value Std. Error   t value
OTUStreptococcus spp.        -6.29e-05      0.207 -0.000303
OTUCampylobacter concisus    -6.28e-05      0.207 -0.000303
OTUPrevotella melaninogenica -6.32e-05      0.207 -0.000305

Intercepts:
       Value  Std. Error t value
0|I    -0.944  0.156     -6.049 
I|II   -0.241  0.151     -1.595 
II|III  1.266  0.161      7.875 
III|IV  2.639  0.207     12.741 

Residual Deviance: 1781.40 
AIC: 1795.40 </code></pre>
<pre class="r"><code># obtain approximate p-values
ctable &lt;- coef(summary(fit))
p &lt;- pnorm(abs(ctable[, &quot;t value&quot;]), lower.tail = FALSE) * 2
(ctable &lt;- cbind(ctable, &quot;p value&quot; = p))</code></pre>
<pre><code>                                  Value Std. Error    t value   p value
OTUStreptococcus spp.        -6.291e-05     0.2073 -0.0003035 9.998e-01
OTUCampylobacter concisus    -6.281e-05     0.2073 -0.0003030 9.998e-01
OTUPrevotella melaninogenica -6.318e-05     0.2073 -0.0003048 9.998e-01
0|I                          -9.445e-01     0.1562 -6.0485909 1.461e-09
I|II                         -2.412e-01     0.1513 -1.5947287 1.108e-01
II|III                        1.266e+00     0.1607  7.8750312 3.407e-15
III|IV                        2.639e+00     0.2071 12.7409977 3.499e-37</code></pre>
<pre class="r"><code># obtain CIs
(ci &lt;- confint(fit))  # CIs assuming normality</code></pre>
<pre><code>Waiting for profiling to be done...</code></pre>
<pre><code>                               2.5 % 97.5 %
OTUStreptococcus spp.        -0.4065 0.4065
OTUCampylobacter concisus    -0.4065 0.4065
OTUPrevotella melaninogenica -0.4065 0.4065</code></pre>
<pre class="r"><code>## OR and CI
exp(cbind(OR = coef(fit), ci))</code></pre>
<pre><code>                                 OR  2.5 % 97.5 %
OTUStreptococcus spp.        0.9999 0.6659  1.502
OTUCampylobacter concisus    0.9999 0.6659  1.502
OTUPrevotella melaninogenica 0.9999 0.6659  1.502</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
# preditive data
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;))
predict(fit, newdata = dotu, &quot;probs&quot;) # only TINY differences</code></pre>
<pre><code>     0    I   II    III      IV
1 0.28 0.16 0.34 0.1533 0.06667
2 0.28 0.16 0.34 0.1533 0.06667
3 0.28 0.16 0.34 0.1533 0.06667
4 0.28 0.16 0.34 0.1533 0.06667</code></pre>
<pre class="r"><code>## store the predicted probabilities for each value of ses
pp.otu &lt;-cbind(dotu, predict(fit, newdata = dotu, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.otu[, 2:6], pp.otu$OTU, colMeans)</code></pre>
<pre><code>pp.otu$OTU: Campylobacter concisus
      0       I      II     III      IV 
0.28001 0.16000 0.34000 0.15333 0.06667 
------------------------------------------------------------ 
pp.otu$OTU: Fusobacterium nucleatum
      0       I      II     III      IV 
0.27999 0.15999 0.34001 0.15334 0.06667 
------------------------------------------------------------ 
pp.otu$OTU: Prevotella melaninogenica
      0       I      II     III      IV 
0.28001 0.16000 0.34000 0.15333 0.06667 
------------------------------------------------------------ 
pp.otu$OTU: Streptococcus spp.
      0       I      II     III      IV 
0.28001 0.16000 0.34000 0.15333 0.06667 </code></pre>
</div>
<div id="model-2-ts-abundance" class="section level3">
<h3>Model 2: TS ~ Abundance</h3>
<pre class="r"><code>## fit ordered logit model and store results
fit &lt;- MASS::polr(tumor.stage ~ Abundance, data = dat.16s.s, Hess=TRUE)

## view a summary of the model
summary(fit)</code></pre>
<pre><code>Call:
MASS::polr(formula = tumor.stage ~ Abundance, data = dat.16s.s, 
    Hess = TRUE)

Coefficients:
            Value Std. Error t value
Abundance 0.00203    0.00384   0.529

Intercepts:
       Value  Std. Error t value
0|I    -0.925  0.098     -9.432 
I|II   -0.221  0.091     -2.442 
II|III  1.286  0.106     12.159 
III|IV  2.659  0.168     15.829 

Residual Deviance: 1781.12 
AIC: 1791.12 </code></pre>
<pre class="r"><code># obtain approximate p-values
ctable &lt;- coef(summary(fit))
p &lt;- pnorm(abs(ctable[, &quot;t value&quot;]), lower.tail = FALSE) * 2
(ctable &lt;- cbind(ctable, &quot;p value&quot; = p))</code></pre>
<pre><code>              Value Std. Error t value   p value
Abundance  0.002029   0.003837  0.5286 5.971e-01
0|I       -0.924970   0.098068 -9.4319 4.027e-21
I|II      -0.221123   0.090546 -2.4421 1.460e-02
II|III     1.285730   0.105746 12.1587 5.160e-34
III|IV     2.658753   0.167964 15.8293 1.953e-56</code></pre>
<pre class="r"><code># obtain CIs
(ci &lt;- confint(fit))  # CIs assuming normality</code></pre>
<pre><code>Waiting for profiling to be done...</code></pre>
<pre><code>    2.5 %    97.5 % 
-0.005522  0.009549 </code></pre>
<pre class="r"><code>## OR and CI
exp(cbind(OR = coef(fit), ci))</code></pre>
<pre><code>          OR     ci
2.5 %  1.002 0.9945
97.5 % 1.002 1.0096</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), Abundance = mean(dat.16s.s$Abundance))
predict(fit, newdata = dotu, &quot;probs&quot;) # bigger differences</code></pre>
<pre><code>       0      I     II    III      IV
1 0.2801 0.1602 0.3399 0.1532 0.06662
2 0.2801 0.1602 0.3399 0.1532 0.06662
3 0.2801 0.1602 0.3399 0.1532 0.06662
4 0.2801 0.1602 0.3399 0.1532 0.06662</code></pre>
<pre class="r"><code>## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU
dabund &lt;- data.frame(
  OTU = rep(c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), each = 51),
  Abundance = rep(seq(0, 500,10), 4)
)
pp.abund &lt;-cbind(dabund, predict(fit, newdata = dabund, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.abund[, 3:7], pp.abund$OTU, colMeans)</code></pre>
<pre><code>pp.abund$OTU: Campylobacter concisus
     0      I     II    III     IV 
0.1970 0.1319 0.3530 0.2106 0.1075 
------------------------------------------------------------ 
pp.abund$OTU: Fusobacterium nucleatum
     0      I     II    III     IV 
0.1970 0.1319 0.3530 0.2106 0.1075 
------------------------------------------------------------ 
pp.abund$OTU: Prevotella melaninogenica
     0      I     II    III     IV 
0.1970 0.1319 0.3530 0.2106 0.1075 
------------------------------------------------------------ 
pp.abund$OTU: Streptococcus spp.
     0      I     II    III     IV 
0.1970 0.1319 0.3530 0.2106 0.1075 </code></pre>
<pre class="r"><code>## melt data set to long for ggplot2
lpp &lt;- melt(pp.abund, id.vars = c(&quot;OTU&quot;, &quot;Abundance&quot;), value.name = &quot;probability&quot;)

## plot predicted probabilities across Abundance values for each level of OTU
## facetted by tumor.stage
ggplot(lpp, aes(x = Abundance, y = probability)) +
  geom_line() + 
  facet_grid(variable ~., scales=&quot;free&quot;)+
  labs(y=&quot;Probability of Tumor Stage&quot;,
       title=&quot;Tumor stage likelihood with bacteria abundance&quot;)+
  theme(
    panel.grid = element_blank()
  )</code></pre>
<p><img src="figure/ordered-logit-model.Rmd/glm-16s-m2-1.png" width="225%" height="10" style="display: block; margin: auto;" /></p>
</div>
<div id="model-3-ts-otu-abundance" class="section level3">
<h3>Model 3: TS ~ OTU + Abundance</h3>
<pre class="r"><code>## fit ordered logit model and store results
fit &lt;- MASS::polr(tumor.stage ~ OTU + Abundance, data = dat.16s.s, Hess=TRUE)

## view a summary of the model
summary(fit)</code></pre>
<pre><code>Call:
MASS::polr(formula = tumor.stage ~ OTU + Abundance, data = dat.16s.s, 
    Hess = TRUE)

Coefficients:
                                Value Std. Error t value
OTUStreptococcus spp.        -0.07580    0.23782 -0.3187
OTUCampylobacter concisus     0.01201    0.20806  0.0577
OTUPrevotella melaninogenica -0.00282    0.20726 -0.0136
Abundance                     0.00309    0.00474  0.6527

Intercepts:
       Value  Std. Error t value
0|I    -0.931  0.157     -5.921 
I|II   -0.227  0.153     -1.489 
II|III  1.280  0.162      7.895 
III|IV  2.653  0.208     12.744 

Residual Deviance: 1780.97 
AIC: 1796.97 </code></pre>
<pre class="r"><code># obtain approximate p-values
ctable &lt;- coef(summary(fit))
p &lt;- pnorm(abs(ctable[, &quot;t value&quot;]), lower.tail = FALSE) * 2
(ctable &lt;- cbind(ctable, &quot;p value&quot; = p))</code></pre>
<pre><code>                                 Value Std. Error  t value   p value
OTUStreptococcus spp.        -0.075802   0.237819 -0.31874 7.499e-01
OTUCampylobacter concisus     0.012013   0.208062  0.05774 9.540e-01
OTUPrevotella melaninogenica -0.002818   0.207263 -0.01360 9.892e-01
Abundance                     0.003091   0.004736  0.65267 5.140e-01
0|I                          -0.931388   0.157314 -5.92055 3.209e-09
I|II                         -0.227249   0.152659 -1.48861 1.366e-01
II|III                        1.279684   0.162094  7.89470 2.910e-15
III|IV                        2.652544   0.208139 12.74410 3.363e-37</code></pre>
<pre class="r"><code># obtain CIs
(ci &lt;- confint(fit))  # CIs assuming normality</code></pre>
<pre><code>Waiting for profiling to be done...</code></pre>
<pre><code>                                 2.5 %  97.5 %
OTUStreptococcus spp.        -0.542806 0.39021
OTUCampylobacter concisus    -0.396086 0.41996
OTUPrevotella melaninogenica -0.409287 0.40363
Abundance                    -0.006224 0.01237</code></pre>
<pre class="r"><code>## OR and CI
exp(cbind(OR = coef(fit), ci))</code></pre>
<pre><code>                                 OR  2.5 % 97.5 %
OTUStreptococcus spp.        0.9270 0.5811  1.477
OTUCampylobacter concisus    1.0121 0.6729  1.522
OTUPrevotella melaninogenica 0.9972 0.6641  1.497
Abundance                    1.0031 0.9938  1.012</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
# predit data
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), Abundance = mean(dat.16s.s$Abundance))
predict(fit, newdata = dotu, &quot;probs&quot;) # bigger differences</code></pre>
<pre><code>       0      I     II    III      IV
1 0.2768 0.1595 0.3411 0.1549 0.06763
2 0.2922 0.1628 0.3352 0.1467 0.06301
3 0.2744 0.1589 0.3420 0.1563 0.06840
4 0.2774 0.1596 0.3409 0.1546 0.06746</code></pre>
<pre class="r"><code>## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU
dabund &lt;- data.frame(
  OTU = rep(c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), each = 51),
  Abundance = rep(seq(0, 500,10), 4)
)
pp.abund &lt;-cbind(dabund, predict(fit, newdata = dabund, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.abund[, 3:7], pp.abund$OTU, colMeans)</code></pre>
<pre><code>pp.abund$OTU: Campylobacter concisus
     0      I     II    III     IV 
0.1615 0.1142 0.3400 0.2418 0.1425 
------------------------------------------------------------ 
pp.abund$OTU: Fusobacterium nucleatum
     0      I     II    III     IV 
0.1631 0.1149 0.3404 0.2406 0.1410 
------------------------------------------------------------ 
pp.abund$OTU: Prevotella melaninogenica
     0      I     II    III     IV 
0.1635 0.1150 0.3405 0.2403 0.1407 
------------------------------------------------------------ 
pp.abund$OTU: Streptococcus spp.
     0      I     II    III     IV 
0.1734 0.1194 0.3425 0.2324 0.1323 </code></pre>
<pre class="r"><code>## melt data set to long for ggplot2
lpp &lt;- melt(pp.abund, id.vars = c(&quot;OTU&quot;, &quot;Abundance&quot;), value.name = &quot;probability&quot;)

## plot predicted probabilities across Abundance values for each level of OTU
## facetted by tumor.stage
ggplot(lpp, aes(x = Abundance, y = probability, colour = OTU)) +
  geom_line() + 
  facet_grid(variable ~., scales=&quot;free&quot;, labeller=&quot;label_both&quot;)+
  labs(y=&quot;Probability of Tumor Stage&quot;,
       title=&quot;Tumor stage likelihood with bacteria abundance and OTU&quot;)+
  theme(
    panel.grid = element_blank()
  )</code></pre>
<p><img src="figure/ordered-logit-model.Rmd/glm-16s-m3-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="model-4-ts-otu-abundance-otuabundnace" class="section level3">
<h3>Model 4: TS ~ OTU + Abundance + OTU:Abundnace</h3>
<pre class="r"><code>## fit ordered logit model and store results
fit &lt;- MASS::polr(tumor.stage ~ OTU + Abundance+ OTU:Abundance, data = dat.16s.s, Hess=TRUE)

## view a summary of the model
summary(fit)</code></pre>
<pre><code>Call:
MASS::polr(formula = tumor.stage ~ OTU + Abundance + OTU:Abundance, 
    data = dat.16s.s, Hess = TRUE)

Coefficients:
                                          Value Std. Error t value
OTUStreptococcus spp.                   0.11164     0.2670   0.418
OTUCampylobacter concisus               0.07891     0.2167   0.364
OTUPrevotella melaninogenica           -0.02403     0.2284  -0.105
Abundance                               0.01545     0.0118   1.306
OTUStreptococcus spp.:Abundance        -0.01691     0.0131  -1.294
OTUCampylobacter concisus:Abundance    -0.03294     0.0639  -0.516
OTUPrevotella melaninogenica:Abundance  0.00191     0.0188   0.102

Intercepts:
       Value  Std. Error t value
0|I    -0.880  0.164     -5.363 
I|II   -0.173  0.160     -1.080 
II|III  1.340  0.170      7.863 
III|IV  2.715  0.215     12.642 

Residual Deviance: 1778.15 
AIC: 1800.15 </code></pre>
<pre class="r"><code># obtain approximate p-values
ctable &lt;- coef(summary(fit))
p &lt;- pnorm(abs(ctable[, &quot;t value&quot;]), lower.tail = FALSE) * 2
(ctable &lt;- cbind(ctable, &quot;p value&quot; = p))</code></pre>
<pre><code>                                           Value Std. Error t value   p value
OTUStreptococcus spp.                   0.111638    0.26698  0.4181 6.758e-01
OTUCampylobacter concisus               0.078914    0.21672  0.3641 7.158e-01
OTUPrevotella melaninogenica           -0.024033    0.22839 -0.1052 9.162e-01
Abundance                               0.015445    0.01183  1.3061 1.915e-01
OTUStreptococcus spp.:Abundance        -0.016909    0.01306 -1.2943 1.956e-01
OTUCampylobacter concisus:Abundance    -0.032940    0.06388 -0.5156 6.061e-01
OTUPrevotella melaninogenica:Abundance  0.001913    0.01884  0.1016 9.191e-01
0|I                                    -0.879584    0.16400 -5.3633 8.174e-08
I|II                                   -0.172875    0.16011 -1.0797 2.803e-01
II|III                                  1.340472    0.17047  7.8633 3.740e-15
III|IV                                  2.715144    0.21478 12.6417 1.244e-36</code></pre>
<pre class="r"><code># obtain CIs
(ci &lt;- confint(fit))  # CIs assuming normality</code></pre>
<pre><code>Waiting for profiling to be done...</code></pre>
<pre><code>                                           2.5 %   97.5 %
OTUStreptococcus spp.                  -0.412893 0.634477
OTUCampylobacter concisus              -0.345946 0.504079
OTUPrevotella melaninogenica           -0.472057 0.423783
Abundance                              -0.007727 0.039526
OTUStreptococcus spp.:Abundance        -0.043247 0.008643
OTUCampylobacter concisus:Abundance    -0.169623 0.092063
OTUPrevotella melaninogenica:Abundance -0.035537 0.038820</code></pre>
<pre class="r"><code>## OR and CI
exp(cbind(OR = coef(fit), ci))</code></pre>
<pre><code>                                           OR  2.5 % 97.5 %
OTUStreptococcus spp.                  1.1181 0.6617  1.886
OTUCampylobacter concisus              1.0821 0.7076  1.655
OTUPrevotella melaninogenica           0.9763 0.6237  1.528
Abundance                              1.0156 0.9923  1.040
OTUStreptococcus spp.:Abundance        0.9832 0.9577  1.009
OTUCampylobacter concisus:Abundance    0.9676 0.8440  1.096
OTUPrevotella melaninogenica:Abundance 1.0019 0.9651  1.040</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
# predit data
gmeans &lt;- dat.16s.s %&gt;% group_by(OTU) %&gt;% summarise(M = mean(Abundance))
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), Abundance = gmeans$M)
predict(fit, newdata = dotu, &quot;probs&quot;) # bigger differences</code></pre>
<pre><code>       0      I     II    III      IV
1 0.2815 0.1612 0.3403 0.1515 0.06552
2 0.2788 0.1606 0.3413 0.1530 0.06633
3 0.2788 0.1606 0.3413 0.1530 0.06635
4 0.2793 0.1607 0.3411 0.1527 0.06618</code></pre>
<pre class="r"><code>## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU
dabund &lt;- data.frame(
  OTU = rep(c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), each = 51),
  Abundance = rep(seq(0, 500,10), 4)
)
pp.abund &lt;-cbind(dabund, predict(fit, newdata = dabund, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.abund[, 3:7], pp.abund$OTU, colMeans)</code></pre>
<pre><code>pp.abund$OTU: Campylobacter concisus
       0        I       II      III       IV 
0.849088 0.052662 0.068093 0.021732 0.008426 
------------------------------------------------------------ 
pp.abund$OTU: Fusobacterium nucleatum
      0       I      II     III      IV 
0.04697 0.03502 0.12530 0.15395 0.63875 
------------------------------------------------------------ 
pp.abund$OTU: Prevotella melaninogenica
      0       I      II     III      IV 
0.04299 0.03181 0.11282 0.13782 0.67457 
------------------------------------------------------------ 
pp.abund$OTU: Streptococcus spp.
      0       I      II     III      IV 
0.35013 0.16995 0.30904 0.12107 0.04981 </code></pre>
<pre class="r"><code>## melt data set to long for ggplot2
lpp &lt;- melt(pp.abund, id.vars = c(&quot;OTU&quot;, &quot;Abundance&quot;), value.name = &quot;probability&quot;) %&gt;%
  mutate(Tumor_Stage = variable)

## plot predicted probabilities across Abundance values for each level of OTU
## facetted by tumor.stage
ggplot(lpp, aes(x = Abundance, y = probability, colour = OTU)) +
  geom_line() + 
  facet_grid(Tumor_Stage ~., scales=&quot;free&quot;, labeller=&quot;label_both&quot;)+
  labs(y=&quot;Probability of Tumor Stage&quot;,
       title=&quot;Tumor stage likelihood with bacteria abundance and OTU&quot;)+
  theme(
    panel.grid = element_blank()
  )</code></pre>
<p><img src="figure/ordered-logit-model.Rmd/glm-16s-m4-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
<pre class="r"><code># proportional odds assumption
glm(I(as.numeric(tumor.stage) &gt;= 2) ~ OTU, family=&quot;binomial&quot;, data = dat.16s.s)</code></pre>
<pre><code>
Call:  glm(formula = I(as.numeric(tumor.stage) &gt;= 2) ~ OTU, family = &quot;binomial&quot;, 
    data = dat.16s.s)

Coefficients:
                 (Intercept)         OTUStreptococcus spp.  
                    9.44e-01                     -1.38e-15  
   OTUCampylobacter concisus  OTUPrevotella melaninogenica  
                   -5.28e-16                     -3.14e-16  

Degrees of Freedom: 599 Total (i.e. Null);  596 Residual
Null Deviance:      712 
Residual Deviance: 712  AIC: 720</code></pre>
<pre class="r"><code>glm(I(as.numeric(tumor.stage) &gt;= 3) ~ OTU, family=&quot;binomial&quot;, data = dat.16s.s)</code></pre>
<pre><code>
Call:  glm(formula = I(as.numeric(tumor.stage) &gt;= 3) ~ OTU, family = &quot;binomial&quot;, 
    data = dat.16s.s)

Coefficients:
                 (Intercept)         OTUStreptococcus spp.  
                    2.41e-01                     -1.88e-15  
   OTUCampylobacter concisus  OTUPrevotella melaninogenica  
                   -2.70e-15                     -1.73e-15  

Degrees of Freedom: 599 Total (i.e. Null);  596 Residual
Null Deviance:      823 
Residual Deviance: 823  AIC: 831</code></pre>
<pre class="r"><code>glm(I(as.numeric(tumor.stage) &gt;= 4) ~ OTU, family=&quot;binomial&quot;, data = dat.16s.s)</code></pre>
<pre><code>
Call:  glm(formula = I(as.numeric(tumor.stage) &gt;= 4) ~ OTU, family = &quot;binomial&quot;, 
    data = dat.16s.s)

Coefficients:
                 (Intercept)         OTUStreptococcus spp.  
                   -1.27e+00                     -5.08e-15  
   OTUCampylobacter concisus  OTUPrevotella melaninogenica  
                   -7.88e-16                     -3.16e-15  

Degrees of Freedom: 599 Total (i.e. Null);  596 Residual
Null Deviance:      632 
Residual Deviance: 632  AIC: 640</code></pre>
<pre class="r"><code>glm(I(as.numeric(tumor.stage) &gt;= 5) ~ OTU, family=&quot;binomial&quot;, data = dat.16s.s)</code></pre>
<pre><code>
Call:  glm(formula = I(as.numeric(tumor.stage) &gt;= 5) ~ OTU, family = &quot;binomial&quot;, 
    data = dat.16s.s)

Coefficients:
                 (Intercept)         OTUStreptococcus spp.  
                   -2.64e+00                      8.92e-16  
   OTUCampylobacter concisus  OTUPrevotella melaninogenica  
                    2.04e-15                      3.21e-16  

Degrees of Freedom: 599 Total (i.e. Null);  596 Residual
Null Deviance:      294 
Residual Deviance: 294  AIC: 302</code></pre>
<!-- # TCGA RNAseq data -->
<!-- ## Double Checking Data -->
<!-- ```{r rna-fig1, fig.height=5, out.width="100%", out.height="100%"} -->
<!-- # in long format -->
<!-- table(dat.rna$tumor.stage) -->
<!-- # by subject -->
<!-- dat <- dat.rna %>% filter(OTU == "Fusobacterium nucleatum") -->
<!-- table(dat$tumor.stage) -->
<!-- sum(table(dat$tumor.stage)) # sample size met -->
<!-- dat.rna.s$Abundancej <- dat.rna.s$Abundance+0.01 -->
<!-- mean.dat <- dat.rna.s %>% -->
<!--   group_by(tumor.stage, OTU) %>% -->
<!--   summarize(M = mean(Abundance)) -->
<!-- ggplot(dat.rna.s, aes(x=tumor.stage, y=Abundancej))+ -->
<!--   geom_violin()+ -->
<!--   geom_jitter(alpha=0.25,width = 0.25)+ -->
<!--   geom_point(data=mean.dat, aes(x=tumor.stage, y = M), size=2, alpha =0.9, color="red")+ -->
<!--   labs(x="Tumor Stage",  -->
<!--        title="Distribution of abundance across tumor stage", -->
<!--        subtitle="Red dot is average abundnace")+ -->
<!--   scale_y_continuous( -->
<!--     breaks=c(0, 100, 1000, 10000, 100000, 150000), -->
<!--     #limits = c(0,500), -->
<!--     trans="pseudo_log")+ -->
<!--   facet_wrap(.~OTU, nrow=1)+ -->
<!--   theme_classic() -->
<!-- ``` -->
<!-- ## Multinomial Logistic Regression -->
<!-- ### Model 1: TS ~ Abundance -->
<!-- ```{r glm-rna-m1} -->
<!-- fit <- nnet::multinom(tumor.stage ~ OTU, data=dat.rna.s) -->
<!-- summary(fit) -->
<!-- # obtain significance tests -->
<!-- z <- summary(fit)$coefficients/summary(fit)$standard.errors -->
<!-- z -->
<!-- p <- (1 - pnorm(abs(z), 0, 1)) * 2 -->
<!-- p -->
<!-- ## extract the coefficients from the model and exponentiate -->
<!-- exp(coef(fit)) # interpret as relative risks -->
<!-- # save fitted logits -->
<!-- pp <- fitted(fit) -->
<!-- # preditive data -->
<!-- dotu <- data.frame(OTU = c("Fusobacterium nucleatum", "Streptococcus spp.", "Campylobacter concisus", "Prevotella melaninogenica")) -->
<!-- predict(fit, newdata = dotu, "probs") # only TINY differences -->
<!-- ## store the predicted probabilities for each value of ses -->
<!-- pp.otu <-cbind(dotu, predict(fit, newdata = dotu, "probs", se = TRUE)) -->
<!-- ## calculate the mean probabilities within each level of OTU -->
<!-- by(pp.otu[, 2:5], pp.otu$OTU, colMeans) -->
<!-- ``` -->
<!-- ### Model 2: TS ~ OTU + Abundance -->
<!-- ```{r glm-rna-m2, fig.height=5, out.width="100%", out.height="100%"} -->
<!-- fit <- nnet::multinom(tumor.stage ~ OTU + Abundance, data=dat.rna.s) -->
<!-- summary(fit) -->
<!-- # obtain significance tests -->
<!-- z <- summary(fit)$coefficients/summary(fit)$standard.errors -->
<!-- z -->
<!-- p <- (1 - pnorm(abs(z), 0, 1)) * 2 -->
<!-- p -->
<!-- ## extract the coefficients from the model and exponentiate -->
<!-- exp(coef(fit)) # interpret as relative risks -->
<!-- # save fitted logits -->
<!-- pp <- fitted(fit) -->
<!-- # predit data -->
<!-- dotu <- data.frame(OTU = c("Fusobacterium nucleatum", "Streptococcus spp.", "Campylobacter concisus", "Prevotella melaninogenica"), Abundance = mean(dat.rna.s$Abundance)) -->
<!-- predict(fit, newdata = dotu, "probs") # bigger differences -->
<!-- ## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU -->
<!-- dabund <- data.frame( -->
<!--   OTU = rep(c("Fusobacterium nucleatum", "Streptococcus spp.", "Campylobacter concisus", "Prevotella melaninogenica"), each = 51), -->
<!--   Abundance = rep(seq(0, 500,10), 4) -->
<!-- ) -->
<!-- pp.abund <-cbind(dabund, predict(fit, newdata = dabund, "probs", se = TRUE)) -->
<!-- ## calculate the mean probabilities within each level of OTU -->
<!-- by(pp.abund[, 3:6], pp.abund$OTU, colMeans) -->
<!-- ## melt data set to long for ggplot2 -->
<!-- lpp <- melt(pp.abund, id.vars = c("OTU", "Abundance"), value.name = "probability") -->
<!-- ## plot predicted probabilities across Abundance values for each level of OTU -->
<!-- ## facetted by tumor.stage -->
<!-- ggplot(lpp, aes(x = Abundance, y = probability, colour = OTU)) + -->
<!--   geom_line() +  -->
<!--   facet_grid(variable ~., scales="free")+ -->
<!--   labs(y="Probability of Tumor Stage", -->
<!--        title="Tumor stage likelihood with bacteria abundance and OTU")+ -->
<!--   theme( -->
<!--     panel.grid = element_blank() -->
<!--   ) -->
<!-- ``` -->
<!-- ### Model 3: TS ~ OTU + Abundance + OTU:Abundnace -->
<!-- ```{r glm-rna-m3, fig.height=5, out.width="100%", out.height="100%"} -->
<!-- fit <- nnet::multinom(tumor.stage ~ OTU + Abundance + OTU:Abundance, data=dat.rna.s) -->
<!-- summary(fit) -->
<!-- # obtain significance tests -->
<!-- z <- summary(fit)$coefficients/summary(fit)$standard.errors -->
<!-- z -->
<!-- p <- (1 - pnorm(abs(z), 0, 1)) * 2 -->
<!-- p -->
<!-- ## extract the coefficients from the model and exponentiate -->
<!-- exp(coef(fit)) # interpret as relative risks -->
<!-- # save fitted logits -->
<!-- pp <- fitted(fit) -->
<!-- # predit data -->
<!-- gmeans <- dat.rna.s %>% group_by(OTU) %>% summarise(M = mean(Abundance)) -->
<!-- dotu <- data.frame(OTU = c("Fusobacterium nucleatum", "Streptococcus spp.", "Campylobacter concisus", "Prevotella melaninogenica"), Abundance = gmeans$M) -->
<!-- predict(fit, newdata = dotu, "probs") # bigger differences -->
<!-- ## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU -->
<!-- dabund <- data.frame( -->
<!--   OTU = rep(c("Fusobacterium nucleatum", "Streptococcus spp.", "Campylobacter concisus", "Prevotella melaninogenica"), each = 51), -->
<!--   Abundance = rep(seq(0, 500,10), 4) -->
<!-- ) -->
<!-- pp.abund <-cbind(dabund, predict(fit, newdata = dabund, "probs", se = TRUE)) -->
<!-- ## calculate the mean probabilities within each level of OTU -->
<!-- by(pp.abund[, 3:6], pp.abund$OTU, colMeans) -->
<!-- ## melt data set to long for ggplot2 -->
<!-- lpp <- melt(pp.abund, id.vars = c("OTU", "Abundance"), value.name = "probability") -->
<!-- ## plot predicted probabilities across Abundance values for each level of OTU -->
<!-- ## facetted by tumor.stage -->
<!-- ggplot(lpp, aes(x = Abundance, y = probability, colour = OTU)) + -->
<!--   geom_line() +  -->
<!--   facet_grid(variable ~., scales="free")+ -->
<!--   labs(y="Probability of Tumor Stage", -->
<!--        title="Tumor stage likelihood with bacteria abundance and OTU")+ -->
<!--   theme( -->
<!--     panel.grid = element_blank() -->
<!--   ) -->
<!-- ``` -->
<!-- # TCGA WGS data -->
<!-- ## Double Checking Data -->
<!-- ```{r wgs-fig1, fig.height=5, out.width="100%", out.height="100%"} -->
<!-- # in long format -->
<!-- table(dat.wgs$tumor.stage) -->
<!-- # by subject -->
<!-- dat <- dat.wgs %>% filter(OTU == "Fusobacterium nucleatum") -->
<!-- table(dat$tumor.stage) -->
<!-- sum(table(dat$tumor.stage)) # sample size met -->
<!-- dat.wgs.s$Abundancej <- dat.wgs.s$Abundance+0.01 -->
<!-- mean.dat <- dat.wgs.s %>% -->
<!--   group_by(tumor.stage, OTU) %>% -->
<!--   summarize(M = mean(Abundance)) -->
<!-- ggplot(dat.wgs.s, aes(x=tumor.stage, y=Abundancej))+ -->
<!--   geom_violin()+ -->
<!--   geom_jitter(alpha=0.25,width = 0.25)+ -->
<!--   geom_point(data=mean.dat, aes(x=tumor.stage, y = M), size=2, alpha =0.9, color="red")+ -->
<!--   labs(x="Tumor Stage",  -->
<!--        title="Distribution of abundance across tumor stage", -->
<!--        subtitle="Red dot is average abundnace")+ -->
<!--   scale_y_continuous( -->
<!--     breaks=c(0, 100, 1000, 10000, 100000), -->
<!--     #limits = c(0,500), -->
<!--     trans="pseudo_log")+ -->
<!--   facet_wrap(.~OTU, nrow=1)+ -->
<!--   theme_classic() -->
<!-- ``` -->
<!-- ## Multinomial Logistic Regression -->
<!-- ### Model 1: TS ~ Abundance -->
<!-- ```{r glm-wgs-m1} -->
<!-- fit <- nnet::multinom(tumor.stage ~ OTU, data=dat.wgs.s) -->
<!-- summary(fit) -->
<!-- # obtain significance tests -->
<!-- z <- summary(fit)$coefficients/summary(fit)$standard.errors -->
<!-- z -->
<!-- p <- (1 - pnorm(abs(z), 0, 1)) * 2 -->
<!-- p -->
<!-- ## extract the coefficients from the model and exponentiate -->
<!-- exp(coef(fit)) # interpret as relative risks -->
<!-- # save fitted logits -->
<!-- pp <- fitted(fit) -->
<!-- # preditive data -->
<!-- dotu <- data.frame(OTU = c("Fusobacterium nucleatum", "Streptococcus spp.", "Campylobacter concisus", "Prevotella melaninogenica")) -->
<!-- predict(fit, newdata = dotu, "probs") # only TINY differences -->
<!-- ## store the predicted probabilities for each value of ses -->
<!-- pp.otu <-cbind(dotu, predict(fit, newdata = dotu, "probs", se = TRUE)) -->
<!-- ## calculate the mean probabilities within each level of OTU -->
<!-- by(pp.otu[, 2:5], pp.otu$OTU, colMeans) -->
<!-- ``` -->
<!-- ### Model 2: TS ~ OTU + Abundance -->
<!-- ```{r glm-wgs-m2, fig.height=5, out.width="100%", out.height="100%"} -->
<!-- fit <- nnet::multinom(tumor.stage ~ OTU + Abundance, data=dat.wgs.s) -->
<!-- summary(fit) -->
<!-- # obtain significance tests -->
<!-- z <- summary(fit)$coefficients/summary(fit)$standard.errors -->
<!-- z -->
<!-- p <- (1 - pnorm(abs(z), 0, 1)) * 2 -->
<!-- p -->
<!-- ## extract the coefficients from the model and exponentiate -->
<!-- exp(coef(fit)) # interpret as relative risks -->
<!-- # save fitted logits -->
<!-- pp <- fitted(fit) -->
<!-- # predit data -->
<!-- dotu <- data.frame(OTU = c("Fusobacterium nucleatum", "Streptococcus spp.", "Campylobacter concisus", "Prevotella melaninogenica"), Abundance = mean(dat.wgs.s$Abundance)) -->
<!-- predict(fit, newdata = dotu, "probs") # bigger differences -->
<!-- ## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU -->
<!-- dabund <- data.frame( -->
<!--   OTU = rep(c("Fusobacterium nucleatum", "Streptococcus spp.", "Campylobacter concisus", "Prevotella melaninogenica"), each = 51), -->
<!--   Abundance = rep(seq(0, 500,10), 4) -->
<!-- ) -->
<!-- pp.abund <-cbind(dabund, predict(fit, newdata = dabund, "probs", se = TRUE)) -->
<!-- ## calculate the mean probabilities within each level of OTU -->
<!-- by(pp.abund[, 3:6], pp.abund$OTU, colMeans) -->
<!-- ## melt data set to long for ggplot2 -->
<!-- lpp <- melt(pp.abund, id.vars = c("OTU", "Abundance"), value.name = "probability") -->
<!-- ## plot predicted probabilities across Abundance values for each level of OTU -->
<!-- ## facetted by tumor.stage -->
<!-- ggplot(lpp, aes(x = Abundance, y = probability, colour = OTU)) + -->
<!--   geom_line() +  -->
<!--   facet_grid(variable ~., scales="free")+ -->
<!--   labs(y="Probability of Tumor Stage", -->
<!--        title="Tumor stage likelihood with bacteria abundance and OTU")+ -->
<!--   theme( -->
<!--     panel.grid = element_blank() -->
<!--   ) -->
<!-- ``` -->
<!-- ### Model 3: TS ~ OTU + Abundance + OTU:Abundnace -->
<!-- ```{r glm-wgs-m3, fig.height=5, out.width="100%", out.height="100%"} -->
<!-- fit <- nnet::multinom(tumor.stage ~ OTU + Abundance + OTU:Abundance, data=dat.wgs.s) -->
<!-- summary(fit) -->
<!-- # obtain significance tests -->
<!-- z <- summary(fit)$coefficients/summary(fit)$standard.errors -->
<!-- z -->
<!-- p <- (1 - pnorm(abs(z), 0, 1)) * 2 -->
<!-- p -->
<!-- ## extract the coefficients from the model and exponentiate -->
<!-- exp(coef(fit)) # interpret as relative risks -->
<!-- # save fitted logits -->
<!-- pp <- fitted(fit) -->
<!-- # predit data -->
<!-- gmeans <- dat.wgs.s %>% group_by(OTU) %>% summarise(M = mean(Abundance)) -->
<!-- dotu <- data.frame(OTU = c("Fusobacterium nucleatum", "Streptococcus spp.", "Campylobacter concisus", "Prevotella melaninogenica"), Abundance = gmeans$M) -->
<!-- predict(fit, newdata = dotu, "probs") # bigger differences -->
<!-- ## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU -->
<!-- dabund <- data.frame( -->
<!--   OTU = rep(c("Fusobacterium nucleatum", "Streptococcus spp.", "Campylobacter concisus", "Prevotella melaninogenica"), each = 51), -->
<!--   Abundance = rep(seq(0, 500,10), 4) -->
<!-- ) -->
<!-- pp.abund <-cbind(dabund, predict(fit, newdata = dabund, "probs", se = TRUE)) -->
<!-- ## calculate the mean probabilities within each level of OTU -->
<!-- by(pp.abund[, 3:6], pp.abund$OTU, colMeans) -->
<!-- ## melt data set to long for ggplot2 -->
<!-- lpp <- melt(pp.abund, id.vars = c("OTU", "Abundance"), value.name = "probability") -->
<!-- ## plot predicted probabilities across Abundance values for each level of OTU -->
<!-- ## facetted by tumor.stage -->
<!-- ggplot(lpp, aes(x = Abundance, y = probability, colour = OTU)) + -->
<!--   geom_line() +  -->
<!--   facet_grid(variable ~., scales="free")+ -->
<!--   labs(y="Probability of Tumor Stage", -->
<!--        title="Tumor stage likelihood with bacteria abundance and OTU")+ -->
<!--   theme( -->
<!--     panel.grid = element_blank() -->
<!--   ) -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] cowplot_1.1.1     dendextend_1.14.0 ggdendro_0.1.22   reshape2_1.4.4   
 [5] car_3.0-10        carData_3.0-4     gvlma_1.0.0.3     patchwork_1.1.1  
 [9] viridis_0.5.1     viridisLite_0.3.0 gridExtra_2.3     xtable_1.8-4     
[13] kableExtra_1.3.1  MASS_7.3-53       data.table_1.13.6 readxl_1.3.1     
[17] forcats_0.5.1     stringr_1.4.0     dplyr_1.0.3       purrr_0.3.4      
[21] readr_1.4.0       tidyr_1.1.2       tibble_3.0.6      ggplot2_3.3.3    
[25] tidyverse_1.3.0   lmerTest_3.1-3    lme4_1.1-26       Matrix_1.2-18    
[29] vegan_2.5-7       lattice_0.20-41   permute_0.9-5     phyloseq_1.34.0  
[33] workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] minqa_1.2.4         colorspace_2.0-0    rio_0.5.16         
 [4] ellipsis_0.3.1      rprojroot_2.0.2     XVector_0.30.0     
 [7] fs_1.5.0            rstudioapi_0.13     farver_2.0.3       
[10] lubridate_1.7.9.2   xml2_1.3.2          codetools_0.2-16   
[13] splines_4.0.3       knitr_1.31          ade4_1.7-16        
[16] jsonlite_1.7.2      nloptr_1.2.2.2      broom_0.7.4        
[19] cluster_2.1.0       dbplyr_2.1.0        BiocManager_1.30.10
[22] compiler_4.0.3      httr_1.4.2          backports_1.2.1    
[25] assertthat_0.2.1    cli_2.3.0           later_1.1.0.1      
[28] htmltools_0.5.1.1   prettyunits_1.1.1   tools_4.0.3        
[31] igraph_1.2.6        gtable_0.3.0        glue_1.4.2         
[34] Rcpp_1.0.6          Biobase_2.50.0      cellranger_1.1.0   
[37] vctrs_0.3.6         Biostrings_2.58.0   rhdf5filters_1.2.0 
[40] multtest_2.46.0     ape_5.4-1           nlme_3.1-149       
[43] iterators_1.0.13    xfun_0.20           ps_1.5.0           
[46] openxlsx_4.2.3      rvest_0.3.6         lifecycle_0.2.0    
[49] statmod_1.4.35      zlibbioc_1.36.0     scales_1.1.1       
[52] hms_1.0.0           promises_1.1.1      parallel_4.0.3     
[55] biomformat_1.18.0   rhdf5_2.34.0        curl_4.3           
[58] yaml_2.2.1          stringi_1.5.3       highr_0.8          
[61] S4Vectors_0.28.1    foreach_1.5.1       BiocGenerics_0.36.0
[64] zip_2.1.1           boot_1.3-25         rlang_0.4.10       
[67] pkgconfig_2.0.3     evaluate_0.14       Rhdf5lib_1.12.1    
[70] labeling_0.4.2      tidyselect_1.1.0    plyr_1.8.6         
[73] magrittr_2.0.1      R6_2.5.0            IRanges_2.24.1     
[76] generics_0.1.0      DBI_1.1.1           foreign_0.8-80     
[79] pillar_1.4.7        haven_2.3.1         withr_2.4.1        
[82] mgcv_1.8-33         abind_1.4-5         survival_3.2-7     
[85] modelr_0.1.8        crayon_1.4.1        rmarkdown_2.6      
[88] progress_1.2.2      grid_4.0.3          git2r_0.28.0       
[91] reprex_1.0.0        digest_0.6.27       webshot_0.5.2      
[94] httpuv_1.5.5        numDeriv_2016.8-1.1 stats4_4.0.3       
[97] munsell_0.5.0      </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
