<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Results Output for Question 3</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">esoph-micro-cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Results Output for Question 3</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>esoph-micro-cancer-workflow/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200916code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200916)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200916code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200916)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomGreathouseLabesophmicrocancerworkflowtree1e15f06803a34cd27b680caab83e8407b088ff08targetblank1e15f06a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/tree/1e15f06803a34cd27b680caab83e8407b088ff08" target="_blank">1e15f06</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomGreathouseLabesophmicrocancerworkflowtree1e15f06803a34cd27b680caab83e8407b088ff08targetblank1e15f06a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/tree/1e15f06803a34cd27b680caab83e8407b088ff08" target="_blank">1e15f06</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    data/

Untracked files:
    Untracked:  analysis/species-sample-type-combined.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd
    Modified:   analysis/results-question-3.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/results-question-3.Rmd</code>) and HTML (<code>docs/results-question-3.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/blob/1e15f06803a34cd27b680caab83e8407b088ff08/analysis/results-question-3.Rmd" target="_blank">1e15f06</a>
</td>
<td>
noah-padgett
</td>
<td>
2021-02-04
</td>
<td>
updated multinomial regression results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/GreathouseLab/esoph-micro-cancer-workflow/1e15f06803a34cd27b680caab83e8407b088ff08/docs/results-question-3.html" target="_blank">1e15f06</a>
</td>
<td>
noah-padgett
</td>
<td>
2021-02-04
</td>
<td>
updated multinomial regression results
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/blob/b9a498d47958ed8a600ed466282f02bc89d106a7/analysis/results-question-3.Rmd" target="_blank">b9a498d</a>
</td>
<td>
noah-padgett
</td>
<td>
2021-01-26
</td>
<td>
fixed issue with subsetting
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/GreathouseLab/esoph-micro-cancer-workflow/b9a498d47958ed8a600ed466282f02bc89d106a7/docs/results-question-3.html" target="_blank">b9a498d</a>
</td>
<td>
noah-padgett
</td>
<td>
2021-01-26
</td>
<td>
fixed issue with subsetting
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/blob/65859077bc3495e2e48531d010b0594ebf4089f4/analysis/results-question-3.Rmd" target="_blank">6585907</a>
</td>
<td>
noah-padgett
</td>
<td>
2021-01-21
</td>
<td>
updated figures with dendrogram
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="question-3" class="section level1">
<h1>Question 3</h1>
<pre><code>Q3: Is fuso associated with tumor stage (pTNM) in either data set? Does X bacteria predict stage? Multivariable w/ age, sex, BMI, history of Barrett&#39;s</code></pre>
<p>Add to this analysis:</p>
<ul>
<li>Fusobacterium nucleatum</li>
<li>Streptococcus sanguinis</li>
<li>Campylobacter concisus</li>
<li>Prevotella spp.</li>
</ul>
<p>TCGA drop “not reported” from tumor stage.</p>
</div>
<div id="nci-16s-data" class="section level1">
<h1>NCI 16s data</h1>
<div id="double-checking-data" class="section level2">
<h2>Double Checking Data</h2>
<pre class="r"><code># in long format
table(dat.16s$tumor.stage)</code></pre>
<pre><code>
    0     1     I    II   III    IV 
11088   264  6336 13464  6072  2640 </code></pre>
<pre class="r"><code># by subject
dat &lt;- dat.16s %&gt;% filter(OTU == &quot;Fusobacterium_nucleatum&quot;)
table(dat$tumor.stage)</code></pre>
<pre><code>
  0   1   I  II III  IV 
 42   1  24  51  23  10 </code></pre>
<pre class="r"><code>sum(table(dat$tumor.stage)) # sample size met</code></pre>
<pre><code>[1] 151</code></pre>
<pre class="r"><code>mean.dat &lt;- dat.16s.s %&gt;%
  group_by(tumor.stage, OTU) %&gt;%
  summarize(M = mean(Abundance))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;tumor.stage&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>ggplot(dat.16s.s, aes(x=tumor.stage, y=Abundance))+
  geom_violin()+
  geom_jitter(alpha=0.25,width = 0.25)+
  geom_point(data=mean.dat, aes(x=tumor.stage, y = M), size=2, alpha =0.9, color=&quot;red&quot;)+
  labs(x=&quot;Tumor Stage&quot;, 
       title=&quot;Distribution of abundance across tumor stage&quot;,
       subtitle=&quot;Red dot is average abundnace&quot;)+
  scale_y_continuous(trans=&quot;pseudo_log&quot;)+
  #  breaks=c(0, 10, 100, 200, 300, 400, 500),
  #  limits = c(0,500),
  #  
  facet_wrap(.~OTU, nrow=1, scales=&quot;free&quot;)+
  theme_classic()</code></pre>
<p><img src="figure/results-question-3.Rmd/nci-fig1-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
<p>Stage “1” has only 1 unique sample and will be dropped from subsequent analyses. And remove NA values.</p>
<pre class="r"><code>dat.16s.s &lt;- dat.16s.s %&gt;%
  filter(tumor.stage != &quot;1&quot;)%&gt;%
  mutate(tumor.stage = droplevels(tumor.stage, exclude=c(&quot;1&quot;,NA)))</code></pre>
</div>
<div id="multinomial-logistic-regression" class="section level2">
<h2>Multinomial Logistic Regression</h2>
<div id="model-1-ts-abundance" class="section level3">
<h3>Model 1: TS ~ Abundance</h3>
<pre class="r"><code>fit &lt;- nnet::multinom(tumor.stage ~ OTU, data=dat.16s.s)</code></pre>
<pre><code># weights:  25 (16 variable)
initial  value 965.662747 
iter  10 value 890.715125
final  value 890.698212 
converged</code></pre>
<pre class="r"><code>summary(fit)</code></pre>
<pre><code>Call:
nnet::multinom(formula = tumor.stage ~ OTU, data = dat.16s.s)

Coefficients:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I    -0.5596182         -1.939649e-05             -2.975175e-06
II    0.1941590          3.455919e-05             -9.943881e-06
III  -0.6021807         -1.686834e-05             -3.080810e-06
IV   -1.4350638         -4.487214e-05             -4.141397e-05
    OTUPrevotella melaninogenica
I                   1.219733e-05
II                 -5.417214e-05
III                 1.171972e-05
IV                 -3.796134e-05

Std. Errors:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I     0.2558835             0.3618753                 0.3618735
II    0.2083683             0.2946764                 0.2946771
III   0.2593995             0.3668476                 0.3668458
IV    0.3518630             0.4976144                 0.4976128
    OTUPrevotella melaninogenica
I                      0.3618717
II                     0.2946779
III                    0.3668440
IV                     0.4976112

Residual Deviance: 1781.396 
AIC: 1813.396 </code></pre>
<pre class="r"><code># obtain significance tests
z &lt;- summary(fit)$coefficients/summary(fit)$standard.errors
z</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I    -2.1870038         -5.359994e-05             -8.221589e-06
II    0.9318068          1.172784e-04             -3.374500e-05
III  -2.3214412         -4.598187e-05             -8.398106e-06
IV   -4.0784727         -9.017453e-05             -8.322530e-05
    OTUPrevotella melaninogenica
I                   3.370623e-05
II                 -1.838351e-04
III                 3.194742e-05
IV                 -7.628716e-05</code></pre>
<pre class="r"><code>p &lt;- (1 - pnorm(abs(z), 0, 1)) * 2
p</code></pre>
<pre><code>     (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I   2.874225e-02             0.9999572                 0.9999934
II  3.514364e-01             0.9999064                 0.9999731
III 2.026304e-02             0.9999633                 0.9999933
IV  4.533253e-05             0.9999281                 0.9999336
    OTUPrevotella melaninogenica
I                      0.9999731
II                     0.9998533
III                    0.9999745
IV                     0.9999391</code></pre>
<pre class="r"><code>## extract the coefficients from the model and exponentiate
exp(coef(fit)) # interpret as relative risks</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I     0.5714272             0.9999806                 0.9999970
II    1.2142894             1.0000346                 0.9999901
III   0.5476162             0.9999831                 0.9999969
IV    0.2381002             0.9999551                 0.9999586
    OTUPrevotella melaninogenica
I                      1.0000122
II                     0.9999458
III                    1.0000117
IV                     0.9999620</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
# preditive data
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;))
predict(fit, newdata = dotu, &quot;probs&quot;) # only TINY differences</code></pre>
<pre><code>          0         I        II       III         IV
1 0.2799997 0.1599994 0.3400006 0.1533323 0.06666797
2 0.2799988 0.1599958 0.3400113 0.1533293 0.06666477
3 0.2800016 0.1600001 0.3399996 0.1533330 0.06666568
4 0.2800045 0.1600041 0.3399880 0.1533368 0.06666658</code></pre>
<pre class="r"><code>## store the predicted probabilities for each value of ses
pp.otu &lt;-cbind(dotu, predict(fit, newdata = dotu, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.otu[, 2:6], pp.otu$OTU, colMeans)</code></pre>
<pre><code>pp.otu$OTU: Campylobacter concisus
         0          I         II        III         IV 
0.28000165 0.16000008 0.33999964 0.15333295 0.06666568 
------------------------------------------------------------ 
pp.otu$OTU: Fusobacterium nucleatum
         0          I         II        III         IV 
0.27999966 0.15999942 0.34000061 0.15333234 0.06666797 
------------------------------------------------------------ 
pp.otu$OTU: Prevotella melaninogenica
         0          I         II        III         IV 
0.28000448 0.16000412 0.33998804 0.15333677 0.06666658 
------------------------------------------------------------ 
pp.otu$OTU: Streptococcus spp.
         0          I         II        III         IV 
0.27999880 0.15999583 0.34001132 0.15332928 0.06666477 </code></pre>
</div>
<div id="model-2-ts-otu-abundance" class="section level3">
<h3>Model 2: TS ~ OTU + Abundance</h3>
<pre class="r"><code>fit &lt;- nnet::multinom(tumor.stage ~ OTU + Abundance, data=dat.16s.s)</code></pre>
<pre><code># weights:  30 (20 variable)
initial  value 965.662747 
iter  10 value 894.405308
iter  20 value 888.437951
final  value 888.305310 
converged</code></pre>
<pre class="r"><code>summary(fit)</code></pre>
<pre><code>Call:
nnet::multinom(formula = tumor.stage ~ OTU + Abundance, data = dat.16s.s)

Coefficients:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I    -0.6047834           -0.30035344               0.039373096
II    0.1463525           -0.31682370               0.041724283
III  -0.6049882           -0.02017896               0.002330615
IV   -1.4661115           -0.21051068               0.026960717
    OTUPrevotella melaninogenica    Abundance
I                   -0.020932505 0.0025475291
II                  -0.021835924 0.0026689058
III                 -0.001713673 0.0001950689
IV                  -0.015798446 0.0018570425

Std. Errors:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I     0.2581762             0.4212445                 0.3631679
II    0.2102088             0.3420791                 0.2957261
III   0.2609537             0.4173638                 0.3676527
IV    0.3546729             0.5765377                 0.4991427
    OTUPrevotella melaninogenica   Abundance
I                      0.3624405 0.001759452
II                     0.2953331 0.001474358
III                    0.3672344 0.001927145
IV                     0.4981078 0.002413791

Residual Deviance: 1776.611 
AIC: 1816.611 </code></pre>
<pre class="r"><code># obtain significance tests
z &lt;- summary(fit)$coefficients/summary(fit)$standard.errors
z</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I    -2.3425216            -0.7130145               0.108415695
II    0.6962244            -0.9261709               0.141090984
III  -2.3183736            -0.0483486               0.006339176
IV   -4.1337000            -0.3651291               0.054014042
    OTUPrevotella melaninogenica Abundance
I                   -0.057754322 1.4479109
II                  -0.073936603 1.8102152
III                 -0.004666429 0.1012217
IV                  -0.031716924 0.7693468</code></pre>
<pre class="r"><code>p &lt;- (1 - pnorm(abs(z), 0, 1)) * 2
p</code></pre>
<pre><code>     (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I   1.915392e-02             0.4758368                 0.9136660
II  4.862883e-01             0.3543572                 0.8877981
III 2.042903e-02             0.9614384                 0.9949421
IV  3.569693e-05             0.7150150                 0.9569240
    OTUPrevotella melaninogenica  Abundance
I                      0.9539443 0.14764197
II                     0.9410608 0.07026243
III                    0.9962767 0.91937448
IV                     0.9746978 0.44168748</code></pre>
<pre class="r"><code>## extract the coefficients from the model and exponentiate
exp(coef(fit)) # interpret as relative risks</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I     0.5461927             0.7405564                  1.040158
II    1.1576041             0.7284592                  1.042607
III   0.5460809             0.9800233                  1.002333
IV    0.2308213             0.8101704                  1.027327
    OTUPrevotella melaninogenica Abundance
I                      0.9792851  1.002551
II                     0.9784008  1.002672
III                    0.9982878  1.000195
IV                     0.9843257  1.001859</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
# predit data
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), Abundance = mean(dat.16s.s$Abundance))
predict(fit, newdata = dotu, &quot;probs&quot;) # bigger differences</code></pre>
<pre><code>          0         I        II       III         IV
1 0.2681362 0.1649697 0.3516266 0.1477649 0.06750260
2 0.3169635 0.1444163 0.3027895 0.1711834 0.06464735
3 0.2619039 0.1676063 0.3580873 0.1446672 0.06773543
4 0.2714817 0.1635681 0.3483242 0.1493524 0.06727357</code></pre>
<pre class="r"><code>## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU
dabund &lt;- data.frame(
  OTU = rep(c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), each = 51),
  Abundance = rep(seq(0, 500,10), 4)
)
pp.abund &lt;-cbind(dabund, predict(fit, newdata = dabund, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.abund[, 3:7], pp.abund$OTU, colMeans)</code></pre>
<pre><code>pp.abund$OTU: Campylobacter concisus
         0          I         II        III         IV 
0.18989671 0.19803248 0.43454900 0.10836285 0.06915896 
------------------------------------------------------------ 
pp.abund$OTU: Fusobacterium nucleatum
         0          I         II        III         IV 
0.19511229 0.19583452 0.42873939 0.11108923 0.06922457 
------------------------------------------------------------ 
pp.abund$OTU: Prevotella melaninogenica
         0          I         II        III         IV 
0.19792008 0.19465222 0.42577802 0.11249991 0.06914977 
------------------------------------------------------------ 
pp.abund$OTU: Streptococcus spp.
         0          I         II        III         IV 
0.23733975 0.17799437 0.38347788 0.13252137 0.06866663 </code></pre>
<pre class="r"><code>## melt data set to long for ggplot2
lpp &lt;- melt(pp.abund, id.vars = c(&quot;OTU&quot;, &quot;Abundance&quot;), value.name = &quot;probability&quot;)

## plot predicted probabilities across Abundance values for each level of OTU
## facetted by tumor.stage
ggplot(lpp, aes(x = Abundance, y = probability, colour = OTU)) +
  geom_line() + 
  facet_grid(variable ~., scales=&quot;free&quot;)+
  labs(y=&quot;Probability of Tumor Stage&quot;,
       title=&quot;Tumor stage likelihood with bacteria abundance and OTU&quot;)+
  theme(
    panel.grid = element_blank()
  )</code></pre>
<p><img src="figure/results-question-3.Rmd/glm-16s-m2-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="model-3-ts-otu-abundance-otuabundnace" class="section level3">
<h3>Model 3: TS ~ OTU + Abundance + OTU:Abundnace</h3>
<pre class="r"><code>fit &lt;- nnet::multinom(tumor.stage ~ OTU + Abundance + OTU:Abundance, data=dat.16s.s)</code></pre>
<pre><code># weights:  45 (32 variable)
initial  value 965.662747 
iter  10 value 941.863626
iter  20 value 901.231110
iter  30 value 879.803956
iter  40 value 878.496840
iter  50 value 877.769589
iter  50 value 877.769583
iter  50 value 877.769583
final  value 877.769583 
converged</code></pre>
<pre class="r"><code>summary(fit)</code></pre>
<pre><code>Call:
nnet::multinom(formula = tumor.stage ~ OTU + Abundance + OTU:Abundance, 
    data = dat.16s.s)

Coefficients:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I   -0.59627225           -0.27271221               -0.02313882
II  -0.04841984            0.08882484                0.20646287
III -0.77491056            0.35263258                0.25421326
IV  -1.07350821           -0.49271665               -0.35516007
    OTUPrevotella melaninogenica    Abundance OTUStreptococcus spp.:Abundance
I                    -0.12547323  0.005306701                    -0.003204521
II                    0.14193593  0.017741150                    -0.016633470
III                   0.06042278  0.015333265                    -0.016845951
IV                   -0.66866510 -0.390337397                     0.391291144
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
I                          0.0181909576                            0.001365729
II                        -0.0007532923                           -0.013204522
III                       -0.1484115828                           -0.010359127
IV                         0.3856873183                            0.400781470

Std. Errors:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I     0.2704470             0.4774130                 0.3781801
II    0.2279863             0.3833642                 0.3126257
III   0.2791141             0.4593983                 0.3875920
IV    0.3763586             0.6462099                 0.5206078
    OTUPrevotella melaninogenica   Abundance OTUStreptococcus spp.:Abundance
I                      0.4006033 0.012397294                     0.012552129
II                     0.3301150 0.009700164                     0.009838977
III                    0.4081087 0.010049130                     0.010290832
IV                     0.5627898 0.370175948                     0.370186103
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
I                            0.02888437                             0.01374055
II                           0.02686202                             0.01111699
III                          0.17149082                             0.01184750
IV                           0.37439040                             0.37023237

Residual Deviance: 1755.539 
AIC: 1819.539 </code></pre>
<pre class="r"><code># obtain significance tests
z &lt;- summary(fit)$coefficients/summary(fit)$standard.errors
z</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I    -2.2047656            -0.5712291               -0.06118464
II   -0.2123805             0.2316983                0.66041547
III  -2.7763223             0.7675965                0.65587854
IV   -2.8523545            -0.7624715               -0.68220277
    OTUPrevotella melaninogenica  Abundance OTUStreptococcus spp.:Abundance
I                     -0.3132107  0.4280531                       -0.255297
II                     0.4299590  1.8289536                       -1.690569
III                    0.1480556  1.5258300                       -1.636986
IV                    -1.1881259 -1.0544645                        1.057012
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
I                            0.62978554                             0.09939408
II                          -0.02804303                            -1.18777856
III                         -0.86542001                            -0.87437230
IV                           1.03017416                             1.08251330</code></pre>
<pre class="r"><code>p &lt;- (1 - pnorm(abs(z), 0, 1)) * 2
p</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I   0.027470545             0.5678444                 0.9512122
II  0.831810220             0.8167723                 0.5089872
III 0.005497769             0.4427269                 0.5119023
IV  0.004339667             0.4457787                 0.4951107
    OTUPrevotella melaninogenica  Abundance OTUStreptococcus spp.:Abundance
I                      0.7541206 0.66861244                      0.79849374
II                     0.6672254 0.06740656                      0.09091913
III                    0.8822989 0.12705220                      0.10163334
IV                     0.2347838 0.29167030                      0.29050612
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
I                             0.5288349                              0.9208254
II                            0.9776278                              0.2349207
III                           0.3868083                              0.3819155
IV                            0.3029283                              0.2790245</code></pre>
<pre class="r"><code>## extract the coefficients from the model and exponentiate
exp(coef(fit)) # interpret as relative risks</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
I     0.5508613             0.7613119                 0.9771268
II    0.9527337             1.0928892                 1.2293221
III   0.4607450             1.4228083                 1.2894468
IV    0.3418073             0.6109644                 0.7010612
    OTUPrevotella melaninogenica Abundance OTUStreptococcus spp.:Abundance
I                      0.8820794 1.0053208                       0.9968006
II                     1.1525028 1.0178995                       0.9835041
III                    1.0622856 1.0154514                       0.9832951
IV                     0.5123921 0.6768285                       1.4788890
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
I                             1.0183574                              1.0013667
II                            0.9992470                              0.9868823
III                           0.8620762                              0.9896943
IV                            1.4706248                              1.4929910</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
# predit data
gmeans &lt;- dat.16s.s %&gt;% group_by(OTU) %&gt;% summarise(M = mean(Abundance))
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), Abundance = gmeans$M)
predict(fit, newdata = dotu, &quot;probs&quot;) # bigger differences</code></pre>
<pre><code>          0         I        II       III           IV
1 0.2819113 0.1714285 0.3737664 0.1728267 6.703098e-05
2 0.2819423 0.1585211 0.3426071 0.1496748 6.725471e-02
3 0.2889414 0.1641112 0.3518175 0.1266238 6.850608e-02
4 0.2771348 0.1607634 0.3432581 0.1547942 6.404947e-02</code></pre>
<pre class="r"><code>## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU
dabund &lt;- data.frame(
  OTU = rep(c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), each = 51),
  Abundance = rep(seq(0, 500,10), 4)
)
pp.abund &lt;-cbind(dabund, predict(fit, newdata = dabund, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.abund[, 3:7], pp.abund$OTU, colMeans)</code></pre>
<pre><code>pp.abund$OTU: Campylobacter concisus
          0           I          II         III          IV 
0.046191605 0.645572197 0.294955139 0.004460403 0.008820657 
------------------------------------------------------------ 
pp.abund$OTU: Fusobacterium nucleatum
          0           I          II         III          IV 
0.056138561 0.047243286 0.709510663 0.185038112 0.002069378 
------------------------------------------------------------ 
pp.abund$OTU: Prevotella melaninogenica
        0         I        II       III        IV 
0.1157777 0.2142867 0.2882000 0.1418134 0.2399222 
------------------------------------------------------------ 
pp.abund$OTU: Streptococcus spp.
         0          I         II        III         IV 
0.26143295 0.18858714 0.35802418 0.12294118 0.06901454 </code></pre>
<pre class="r"><code>## melt data set to long for ggplot2
lpp &lt;- melt(pp.abund, id.vars = c(&quot;OTU&quot;, &quot;Abundance&quot;), value.name = &quot;probability&quot;)

## plot predicted probabilities across Abundance values for each level of OTU
## facetted by tumor.stage
ggplot(lpp, aes(x = Abundance, y = probability, colour = OTU)) +
  geom_line() + 
  facet_grid(variable ~., scales=&quot;free&quot;)+
  labs(y=&quot;Probability of Tumor Stage&quot;,
       title=&quot;Tumor stage likelihood with bacteria abundance and OTU&quot;)+
  theme(
    panel.grid = element_blank()
  )</code></pre>
<p><img src="figure/results-question-3.Rmd/glm-16s-m3-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="tcga-rnaseq-data" class="section level1">
<h1>TCGA RNAseq data</h1>
<div id="double-checking-data-1" class="section level2">
<h2>Double Checking Data</h2>
<pre class="r"><code># in long format
table(dat.rna$tumor.stage)</code></pre>
<pre><code>
    I    II   III    IV 
16359 55309 39729  6232 </code></pre>
<pre class="r"><code># by subject
dat &lt;- dat.rna %&gt;% filter(OTU == &quot;Fusobacterium nucleatum&quot;)
table(dat$tumor.stage)</code></pre>
<pre><code>
  I  II III  IV 
 21  71  51   8 </code></pre>
<pre class="r"><code>sum(table(dat$tumor.stage)) # sample size met</code></pre>
<pre><code>[1] 151</code></pre>
<pre class="r"><code>dat.rna.s$Abundancej &lt;- dat.rna.s$Abundance+0.01
mean.dat &lt;- dat.rna.s %&gt;%
  group_by(tumor.stage, OTU) %&gt;%
  summarize(M = mean(Abundance))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;tumor.stage&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>ggplot(dat.rna.s, aes(x=tumor.stage, y=Abundancej))+
  geom_violin()+
  geom_jitter(alpha=0.25,width = 0.25)+
  geom_point(data=mean.dat, aes(x=tumor.stage, y = M), size=2, alpha =0.9, color=&quot;red&quot;)+
  labs(x=&quot;Tumor Stage&quot;, 
       title=&quot;Distribution of abundance across tumor stage&quot;,
       subtitle=&quot;Red dot is average abundnace&quot;)+
  scale_y_continuous(
    breaks=c(0, 100, 1000, 10000, 100000, 150000),
    #limits = c(0,500),
    trans=&quot;pseudo_log&quot;)+
  facet_wrap(.~OTU, nrow=1)+
  theme_classic()</code></pre>
<p><img src="figure/results-question-3.Rmd/rna-fig1-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="multinomial-logistic-regression-1" class="section level2">
<h2>Multinomial Logistic Regression</h2>
<div id="model-1-ts-abundance-1" class="section level3">
<h3>Model 1: TS ~ Abundance</h3>
<pre class="r"><code>fit &lt;- nnet::multinom(tumor.stage ~ OTU, data=dat.rna.s)</code></pre>
<pre><code># weights:  20 (12 variable)
initial  value 1883.974037 
iter  10 value 1566.684823
final  value 1564.788361 
converged</code></pre>
<pre class="r"><code>summary(fit)</code></pre>
<pre><code>Call:
nnet::multinom(formula = tumor.stage ~ OTU, data = dat.rna.s)

Coefficients:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    1.2181362          1.612699e-05              5.420841e-05
III   0.8872408         -4.489247e-06              1.323503e-04
IV   -0.9655016          4.282208e-04              2.974206e-04
    OTUPrevotella melaninogenica
II                  1.180849e-04
III                 7.358456e-05
IV                  2.958596e-04

Std. Errors:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    0.2483960             0.2682988                 0.3512914
III   0.2592769             0.2800517                 0.3666773
IV    0.4155267             0.4488111                 0.5876245
    OTUPrevotella melaninogenica
II                     0.3512910
III                    0.3666798
IV                     0.5876261

Residual Deviance: 3129.577 
AIC: 3153.577 </code></pre>
<pre class="r"><code># obtain significance tests
z &lt;- summary(fit)$coefficients/summary(fit)$standard.errors
z</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II     4.904009          6.010831e-05              0.0001543118
III    3.421981         -1.603006e-05              0.0003609449
IV    -2.323561          9.541226e-04              0.0005061405
    OTUPrevotella melaninogenica
II                  0.0003361456
III                 0.0002006780
IV                  0.0005034828</code></pre>
<pre class="r"><code>p &lt;- (1 - pnorm(abs(z), 0, 1)) * 2
p</code></pre>
<pre><code>     (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II  9.389999e-07             0.9999520                 0.9998769
III 6.216659e-04             0.9999872                 0.9997120
IV  2.014904e-02             0.9992387                 0.9995962
    OTUPrevotella melaninogenica
II                     0.9997318
III                    0.9998399
IV                     0.9995983</code></pre>
<pre class="r"><code>## extract the coefficients from the model and exponentiate
exp(coef(fit)) # interpret as relative risks</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    3.3808806             1.0000161                  1.000054
III   2.4284199             0.9999955                  1.000132
IV    0.3807921             1.0004283                  1.000297
    OTUPrevotella melaninogenica
II                      1.000118
III                     1.000074
IV                      1.000296</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
# preditive data
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;))
predict(fit, newdata = dotu, &quot;probs&quot;) # only TINY differences</code></pre>
<pre><code>          I        II       III         IV
1 0.1390803 0.4702138 0.3377453 0.05296067
2 0.1390763 0.4702078 0.3377341 0.05298183
3 0.1390683 0.4701989 0.3377610 0.05297187
4 0.1390669 0.4702241 0.3377377 0.05297125</code></pre>
<pre class="r"><code>## store the predicted probabilities for each value of ses
pp.otu &lt;-cbind(dotu, predict(fit, newdata = dotu, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.otu[, 2:5], pp.otu$OTU, colMeans)</code></pre>
<pre><code>pp.otu$OTU: Campylobacter concisus
         I         II        III         IV 
0.13906831 0.47019885 0.33776096 0.05297187 
------------------------------------------------------------ 
pp.otu$OTU: Fusobacterium nucleatum
         I         II        III         IV 
0.13908026 0.47021378 0.33774529 0.05296067 
------------------------------------------------------------ 
pp.otu$OTU: Prevotella melaninogenica
         I         II        III         IV 
0.13906691 0.47022414 0.33773770 0.05297125 
------------------------------------------------------------ 
pp.otu$OTU: Streptococcus spp.
         I         II        III         IV 
0.13907627 0.47020784 0.33773406 0.05298183 </code></pre>
</div>
<div id="model-2-ts-otu-abundance-1" class="section level3">
<h3>Model 2: TS ~ OTU + Abundance</h3>
<pre class="r"><code>fit &lt;- nnet::multinom(tumor.stage ~ OTU + Abundance, data=dat.rna.s)</code></pre>
<pre><code># weights:  24 (15 variable)
initial  value 1883.974037 
iter  10 value 1583.114897
iter  20 value 1561.973672
final  value 1561.966301 
converged</code></pre>
<pre class="r"><code>summary(fit)</code></pre>
<pre><code>Call:
nnet::multinom(formula = tumor.stage ~ OTU + Abundance, data = dat.rna.s)

Coefficients:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    1.2223820          -0.003922936              -0.003953146
III   0.8824898           0.004632690               0.004974597
IV   -1.1199777           0.154052453               0.154872317
    OTUPrevotella melaninogenica     Abundance
II                  -0.003549956 -1.318433e-05
III                  0.004212484  1.075128e-05
IV                   0.151986559  5.025792e-05

Std. Errors:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II   0.07557565            0.09072775                0.08689443
III  0.07960033            0.09603308                0.08480810
IV   0.09845530            0.09426987                0.04516448
    OTUPrevotella melaninogenica    Abundance
II                    0.08695669 8.403456e-05
III                   0.08482469 7.405390e-05
IV                    0.04511175 7.293706e-05

Residual Deviance: 3123.933 
AIC: 3153.933 </code></pre>
<pre class="r"><code># obtain significance tests
z &lt;- summary(fit)$coefficients/summary(fit)$standard.errors
z</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II     16.17428           -0.04323854               -0.04549366
III    11.08651            0.04824056                0.05865710
IV    -11.37549            1.63416424                3.42907305
    OTUPrevotella melaninogenica  Abundance
II                   -0.04082441 -0.1568918
III                   0.04966106  0.1451818
IV                    3.36911230  0.6890589</code></pre>
<pre class="r"><code>p &lt;- (1 - pnorm(abs(z), 0, 1)) * 2
p</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II            0             0.9655114              0.9637138257
III           0             0.9615245              0.9532252334
IV            0             0.1022244              0.0006056465
    OTUPrevotella melaninogenica Abundance
II                   0.967435878 0.8753301
III                  0.960392485 0.8845674
IV                   0.000754107 0.4907862</code></pre>
<pre class="r"><code>## extract the coefficients from the model and exponentiate
exp(coef(fit)) # interpret as relative risks</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    3.3952655             0.9960847                 0.9960547
III   2.4169098             1.0046434                 1.0049870
IV    0.3262871             1.1665521                 1.1675089
    OTUPrevotella melaninogenica Abundance
II                     0.9964563 0.9999868
III                    1.0042214 1.0000108
IV                     1.1641446 1.0000503</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
# predit data
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), Abundance = mean(dat.rna.s$Abundance))
predict(fit, newdata = dotu, &quot;probs&quot;) # bigger differences</code></pre>
<pre><code>          I        II       III         IV
1 0.1400928 0.4747230 0.3391318 0.04605243
2 0.1390656 0.4693972 0.3382085 0.05332866
3 0.1390454 0.4693149 0.3382750 0.05336465
4 0.1390763 0.4696085 0.3380925 0.05322270</code></pre>
<pre class="r"><code>## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU
dabund &lt;- data.frame(
  OTU = rep(c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), each = 51),
  Abundance = rep(seq(0, 500,10), 4)
)
pp.abund &lt;-cbind(dabund, predict(fit, newdata = dabund, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.abund[, 3:6], pp.abund$OTU, colMeans)</code></pre>
<pre><code>pp.abund$OTU: Campylobacter concisus
         I         II        III         IV 
0.13904297 0.46867881 0.33863960 0.05363862 
------------------------------------------------------------ 
pp.abund$OTU: Fusobacterium nucleatum
         I         II        III         IV 
0.14009647 0.47410032 0.33951229 0.04629091 
------------------------------------------------------------ 
pp.abund$OTU: Prevotella melaninogenica
         I         II        III         IV 
0.13907406 0.46897265 0.33845728 0.05349601 
------------------------------------------------------------ 
pp.abund$OTU: Streptococcus spp.
         I         II        III         IV 
0.13906321 0.46876120 0.33857312 0.05360246 </code></pre>
<pre class="r"><code>## melt data set to long for ggplot2
lpp &lt;- melt(pp.abund, id.vars = c(&quot;OTU&quot;, &quot;Abundance&quot;), value.name = &quot;probability&quot;)

## plot predicted probabilities across Abundance values for each level of OTU
## facetted by tumor.stage
ggplot(lpp, aes(x = Abundance, y = probability, colour = OTU)) +
  geom_line() + 
  facet_grid(variable ~., scales=&quot;free&quot;)+
  labs(y=&quot;Probability of Tumor Stage&quot;,
       title=&quot;Tumor stage likelihood with bacteria abundance and OTU&quot;)+
  theme(
    panel.grid = element_blank()
  )</code></pre>
<p><img src="figure/results-question-3.Rmd/glm-rna-m2-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="model-3-ts-otu-abundance-otuabundnace-1" class="section level3">
<h3>Model 3: TS ~ OTU + Abundance + OTU:Abundnace</h3>
<pre class="r"><code>fit &lt;- nnet::multinom(tumor.stage ~ OTU + Abundance + OTU:Abundance, data=dat.rna.s)</code></pre>
<pre><code># weights:  36 (24 variable)
initial  value 1883.974037 
iter  10 value 1755.243497
iter  20 value 1578.233029
iter  30 value 1559.100611
iter  40 value 1557.762012
iter  50 value 1557.755273
final  value 1557.755203 
converged</code></pre>
<pre class="r"><code>summary(fit)</code></pre>
<pre><code>Call:
nnet::multinom(formula = tumor.stage ~ OTU + Abundance + OTU:Abundance, 
    data = dat.rna.s)

Coefficients:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II     1.226420           -0.03665663               -0.05037347
III    0.885271           -0.02687636               -0.07806097
IV    -1.119301            0.24098068                0.07696960
    OTUPrevotella melaninogenica     Abundance OTUStreptococcus spp.:Abundance
II                   -0.05421043 -2.453567e-05                     0.001777190
III                  -0.03401722  4.203709e-06                     0.001771215
IV                    0.10598889  4.488828e-05                    -0.010103368
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
II                           0.04286102                            0.001411664
III                          0.05084790                            0.001207072
IV                           0.05041020                            0.001372158

Std. Errors:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II   0.07750445            0.09342979                0.08889085
III  0.08179267            0.09896120                0.08678082
IV   0.10292173            0.09824740                0.04606603
    OTUPrevotella melaninogenica    Abundance OTUStreptococcus spp.:Abundance
II                    0.08850307 8.013276e-05                     0.002040640
III                   0.08598735 6.363161e-05                     0.002056782
IV                    0.04582294 6.036275e-05                     0.008685766
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
II                           0.07411623                            0.002591894
III                          0.07407726                            0.002613532
IV                           0.07450400                            0.002776801

Residual Deviance: 3115.51 
AIC: 3163.51 </code></pre>
<pre class="r"><code># obtain significance tests
z &lt;- summary(fit)$coefficients/summary(fit)$standard.errors
z</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II     15.82387            -0.3923441                -0.5666891
III    10.82335            -0.2715849                -0.8995187
IV    -10.87527             2.4527945                 1.6708539
    OTUPrevotella melaninogenica   Abundance OTUStreptococcus spp.:Abundance
II                    -0.6125260 -0.30618774                       0.8708982
III                   -0.3956073  0.06606322                       0.8611586
IV                     2.3130093  0.74364206                      -1.1632098
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
II                            0.5782946                              0.5446457
III                           0.6864170                              0.4618546
IV                            0.6766106                              0.4941505</code></pre>
<pre class="r"><code>p &lt;- (1 - pnorm(abs(z), 0, 1)) * 2
p</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II            0            0.69480398                0.57092545
III           0            0.78594124                0.36837644
IV            0            0.01417513                0.09475055
    OTUPrevotella melaninogenica Abundance OTUStreptococcus spp.:Abundance
II                    0.54018980 0.7594617                       0.3838097
III                   0.69239476 0.9473275                       0.3891507
IV                    0.02072213 0.4570931                       0.2447444
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
II                            0.5630652                              0.5859972
III                           0.4924502                              0.6441856
IV                            0.4986530                              0.6211999</code></pre>
<pre class="r"><code>## extract the coefficients from the model and exponentiate
exp(coef(fit)) # interpret as relative risks</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    3.4090035             0.9640071                 0.9508742
III   2.4236412             0.9734816                 0.9249080
IV    0.3265079             1.2724965                 1.0800092
    OTUPrevotella melaninogenica Abundance OTUStreptococcus spp.:Abundance
II                     0.9472328 0.9999755                       1.0017788
III                    0.9665549 1.0000042                       1.0017728
IV                     1.1118095 1.0000449                       0.9899475
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
II                             1.043793                               1.001413
III                            1.052163                               1.001208
IV                             1.051702                               1.001373</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
# predit data
gmeans &lt;- dat.rna.s %&gt;% group_by(OTU) %&gt;% summarise(M = mean(Abundance))
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), Abundance = gmeans$M)
predict(fit, newdata = dotu, &quot;probs&quot;) # bigger differences</code></pre>
<pre><code>          I        II       III         IV
1 0.1409701 0.4672700 0.3433074 0.04845248
2 0.1390843 0.4736103 0.3401810 0.04712435
3 0.1158487 0.4810355 0.3484301 0.05468565
4 0.1341457 0.4731400 0.3394230 0.05329129</code></pre>
<pre class="r"><code>## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU
dabund &lt;- data.frame(
  OTU = rep(c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), each = 51),
  Abundance = rep(seq(0, 500,10), 4)
)
pp.abund &lt;-cbind(dabund, predict(fit, newdata = dabund, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.abund[, 3:6], pp.abund$OTU, colMeans)</code></pre>
<pre><code>pp.abund$OTU: Campylobacter concisus
          I          II         III          IV 
0.008166091 0.195378854 0.698070032 0.098385023 
------------------------------------------------------------ 
pp.abund$OTU: Fusobacterium nucleatum
         I         II        III         IV 
0.13996666 0.47423005 0.33958596 0.04621733 
------------------------------------------------------------ 
pp.abund$OTU: Prevotella melaninogenica
         I         II        III         IV 
0.10945729 0.49277179 0.34194999 0.05582093 
------------------------------------------------------------ 
pp.abund$OTU: Streptococcus spp.
         I         II        III         IV 
0.10335100 0.51437864 0.37145003 0.01082033 </code></pre>
<pre class="r"><code>## melt data set to long for ggplot2
lpp &lt;- melt(pp.abund, id.vars = c(&quot;OTU&quot;, &quot;Abundance&quot;), value.name = &quot;probability&quot;)

## plot predicted probabilities across Abundance values for each level of OTU
## facetted by tumor.stage
ggplot(lpp, aes(x = Abundance, y = probability, colour = OTU)) +
  geom_line() + 
  facet_grid(variable ~., scales=&quot;free&quot;)+
  labs(y=&quot;Probability of Tumor Stage&quot;,
       title=&quot;Tumor stage likelihood with bacteria abundance and OTU&quot;)+
  theme(
    panel.grid = element_blank()
  )</code></pre>
<p><img src="figure/results-question-3.Rmd/glm-rna-m3-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="tcga-wgs-data" class="section level1">
<h1>TCGA WGS data</h1>
<div id="double-checking-data-2" class="section level2">
<h2>Double Checking Data</h2>
<pre class="r"><code># in long format
table(dat.wgs$tumor.stage)</code></pre>
<pre><code>
    I    II   III    IV 
14022 38950 21812  4674 </code></pre>
<pre class="r"><code># by subject
dat &lt;- dat.wgs %&gt;% filter(OTU == &quot;Fusobacterium nucleatum&quot;)
table(dat$tumor.stage)</code></pre>
<pre><code>
  I  II III  IV 
 18  50  28   6 </code></pre>
<pre class="r"><code>sum(table(dat$tumor.stage)) # sample size met</code></pre>
<pre><code>[1] 102</code></pre>
<pre class="r"><code>dat.wgs.s$Abundancej &lt;- dat.wgs.s$Abundance+0.01
mean.dat &lt;- dat.wgs.s %&gt;%
  group_by(tumor.stage, OTU) %&gt;%
  summarize(M = mean(Abundance))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;tumor.stage&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>ggplot(dat.wgs.s, aes(x=tumor.stage, y=Abundancej))+
  geom_violin()+
  geom_jitter(alpha=0.25,width = 0.25)+
  geom_point(data=mean.dat, aes(x=tumor.stage, y = M), size=2, alpha =0.9, color=&quot;red&quot;)+
  labs(x=&quot;Tumor Stage&quot;, 
       title=&quot;Distribution of abundance across tumor stage&quot;,
       subtitle=&quot;Red dot is average abundnace&quot;)+
  scale_y_continuous(
    breaks=c(0, 100, 1000, 10000, 100000),
    #limits = c(0,500),
    trans=&quot;pseudo_log&quot;)+
  facet_wrap(.~OTU, nrow=1)+
  theme_classic()</code></pre>
<p><img src="figure/results-question-3.Rmd/wgs-fig1-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="multinomial-logistic-regression-2" class="section level2">
<h2>Multinomial Logistic Regression</h2>
<div id="model-1-ts-abundance-2" class="section level3">
<h3>Model 1: TS ~ Abundance</h3>
<pre class="r"><code>fit &lt;- nnet::multinom(tumor.stage ~ OTU, data=dat.wgs.s)</code></pre>
<pre><code># weights:  20 (12 variable)
initial  value 1272.618224 
iter  10 value 1083.751350
final  value 1080.603922 
converged</code></pre>
<pre class="r"><code>summary(fit)</code></pre>
<pre><code>Call:
nnet::multinom(formula = tumor.stage ~ OTU, data = dat.wgs.s)

Coefficients:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II     1.021682         -8.537835e-05             -2.849032e-04
III    0.441805         -4.125951e-05              7.676744e-05
IV    -1.098720          2.228180e-04             -6.796120e-04
    OTUPrevotella melaninogenica
II                 -3.004313e-05
III                -1.326913e-04
IV                 -6.925208e-04

Std. Errors:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    0.2748727             0.2968961                 0.3887207
III   0.3021108             0.3263164                 0.4272290
IV    0.4714239             0.5091885                 0.6667526
    OTUPrevotella melaninogenica
II                     0.3887206
III                    0.4272449
IV                     0.6667653

Residual Deviance: 2161.208 
AIC: 2185.208 </code></pre>
<pre class="r"><code># obtain significance tests
z &lt;- summary(fit)$coefficients/summary(fit)$standard.errors
z</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II     3.716927         -0.0002875698             -0.0007329253
III    1.462394         -0.0001264402              0.0001796869
IV    -2.330642          0.0004375944             -0.0010192866
    OTUPrevotella melaninogenica
II                 -7.728721e-05
III                -3.105744e-04
IV                 -1.038627e-03</code></pre>
<pre class="r"><code>p &lt;- (1 - pnorm(abs(z), 0, 1)) * 2
p</code></pre>
<pre><code>     (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II  0.0002016604             0.9997706                 0.9994152
III 0.1436333009             0.9998991                 0.9998566
IV  0.0197722350             0.9996509                 0.9991867
    OTUPrevotella melaninogenica
II                     0.9999383
III                    0.9997522
IV                     0.9991713</code></pre>
<pre class="r"><code>## extract the coefficients from the model and exponentiate
exp(coef(fit)) # interpret as relative risks</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    2.7778632             0.9999146                 0.9997151
III   1.5555124             0.9999587                 1.0000768
IV    0.3332973             1.0002228                 0.9993206
    OTUPrevotella melaninogenica
II                     0.9999700
III                    0.9998673
IV                     0.9993077</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
# preditive data
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;))
predict(fit, newdata = dotu, &quot;probs&quot;) # only TINY differences</code></pre>
<pre><code>          I        II       III         IV
1 0.1764704 0.4902106 0.2745019 0.05881711
2 0.1764775 0.4901884 0.2745016 0.05883257
3 0.1764984 0.4901487 0.2745665 0.05878647
4 0.1764866 0.4902409 0.2744907 0.05878179</code></pre>
<pre class="r"><code>## store the predicted probabilities for each value of ses
pp.otu &lt;-cbind(dotu, predict(fit, newdata = dotu, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.otu[, 2:5], pp.otu$OTU, colMeans)</code></pre>
<pre><code>pp.otu$OTU: Campylobacter concisus
         I         II        III         IV 
0.17649838 0.49014867 0.27456648 0.05878647 
------------------------------------------------------------ 
pp.otu$OTU: Fusobacterium nucleatum
         I         II        III         IV 
0.17647040 0.49021061 0.27450188 0.05881711 
------------------------------------------------------------ 
pp.otu$OTU: Prevotella melaninogenica
         I         II        III         IV 
0.17648661 0.49024092 0.27449068 0.05878179 
------------------------------------------------------------ 
pp.otu$OTU: Streptococcus spp.
         I         II        III         IV 
0.17647747 0.49018841 0.27450156 0.05883257 </code></pre>
</div>
<div id="model-2-ts-otu-abundance-2" class="section level3">
<h3>Model 2: TS ~ OTU + Abundance</h3>
<pre class="r"><code>fit &lt;- nnet::multinom(tumor.stage ~ OTU + Abundance, data=dat.wgs.s)</code></pre>
<pre><code># weights:  24 (15 variable)
initial  value 1272.618224 
iter  10 value 1094.897366
iter  20 value 1078.410457
final  value 1078.272984 
converged</code></pre>
<pre class="r"><code>summary(fit)</code></pre>
<pre><code>Call:
nnet::multinom(formula = tumor.stage ~ OTU + Abundance, data = dat.wgs.s)

Coefficients:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    0.9888923            0.01862978                0.01930877
III   0.3803374            0.04494356                0.04665184
IV   -1.1562146            0.04139047                0.04250580
    OTUPrevotella melaninogenica    Abundance
II                   -0.02200784 0.0002502637
III                  -0.02498400 0.0002714197
IV                   -0.02478676 0.0002694783

Std. Errors:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II   0.04353209            0.03376873               0.003395291
III  0.04884489            0.03802708               0.003824299
IV   0.03381891            0.02628255               0.002661367
    OTUPrevotella melaninogenica    Abundance
II                   0.003254089 0.0002822526
III                  0.003589524 0.0002823707
IV                   0.002530123 0.0002832039

Residual Deviance: 2156.546 
AIC: 2186.546 </code></pre>
<pre class="r"><code># obtain significance tests
z &lt;- summary(fit)$coefficients/summary(fit)$standard.errors
z</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    22.716399             0.5516874                  5.686927
III    7.786636             1.1818828                 12.198797
IV   -34.188404             1.5748271                 15.971415
    OTUPrevotella melaninogenica Abundance
II                     -6.763134 0.8866659
III                    -6.960253 0.9612175
IV                     -9.796660 0.9515345</code></pre>
<pre class="r"><code>p &lt;- (1 - pnorm(abs(z), 0, 1)) * 2
p</code></pre>
<pre><code>     (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II  0.000000e+00             0.5811625              1.293461e-08
III 6.883383e-15             0.2372522              0.000000e+00
IV  0.000000e+00             0.1152964              0.000000e+00
    OTUPrevotella melaninogenica Abundance
II                  1.350386e-11 0.3752588
III                 3.396616e-12 0.3364428
IV                  0.000000e+00 0.3413331</code></pre>
<pre class="r"><code>## extract the coefficients from the model and exponentiate
exp(coef(fit)) # interpret as relative risks</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    2.6882550              1.018804                  1.019496
III   1.4627781              1.045969                  1.047757
IV    0.3146751              1.042259                  1.043422
    OTUPrevotella melaninogenica Abundance
II                     0.9782326  1.000250
III                    0.9753255  1.000271
IV                     0.9755179  1.000270</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
# predit data
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), Abundance = mean(dat.wgs.s$Abundance))
predict(fit, newdata = dotu, &quot;probs&quot;) # bigger differences</code></pre>
<pre><code>          I        II       III         IV
1 0.1694758 0.4984673 0.2733043 0.05875264
2 0.1654359 0.4957350 0.2790534 0.05977577
3 0.1652903 0.4956353 0.2792846 0.05978983
4 0.1727639 0.4970775 0.2717324 0.05842625</code></pre>
<pre class="r"><code>## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU
dabund &lt;- data.frame(
  OTU = rep(c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), each = 51),
  Abundance = rep(seq(0, 500,10), 4)
)
pp.abund &lt;-cbind(dabund, predict(fit, newdata = dabund, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.abund[, 3:6], pp.abund$OTU, colMeans)</code></pre>
<pre><code>pp.abund$OTU: Campylobacter concisus
         I         II        III         IV 
0.16929834 0.49370725 0.27756107 0.05943334 
------------------------------------------------------------ 
pp.abund$OTU: Fusobacterium nucleatum
         I         II        III         IV 
0.17356111 0.49646243 0.27158184 0.05839462 
------------------------------------------------------------ 
pp.abund$OTU: Prevotella melaninogenica
         I         II        III         IV 
0.17691065 0.49503069 0.26999400 0.05806466 
------------------------------------------------------------ 
pp.abund$OTU: Streptococcus spp.
         I         II        III         IV 
0.16944659 0.49380429 0.27733003 0.05941909 </code></pre>
<pre class="r"><code>## melt data set to long for ggplot2
lpp &lt;- melt(pp.abund, id.vars = c(&quot;OTU&quot;, &quot;Abundance&quot;), value.name = &quot;probability&quot;)

## plot predicted probabilities across Abundance values for each level of OTU
## facetted by tumor.stage
ggplot(lpp, aes(x = Abundance, y = probability, colour = OTU)) +
  geom_line() + 
  facet_grid(variable ~., scales=&quot;free&quot;)+
  labs(y=&quot;Probability of Tumor Stage&quot;,
       title=&quot;Tumor stage likelihood with bacteria abundance and OTU&quot;)+
  theme(
    panel.grid = element_blank()
  )</code></pre>
<p><img src="figure/results-question-3.Rmd/glm-wgs-m2-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
</div>
<div id="model-3-ts-otu-abundance-otuabundnace-2" class="section level3">
<h3>Model 3: TS ~ OTU + Abundance + OTU:Abundnace</h3>
<pre class="r"><code>fit &lt;- nnet::multinom(tumor.stage ~ OTU + Abundance + OTU:Abundance, data=dat.wgs.s)</code></pre>
<pre><code># weights:  36 (24 variable)
initial  value 1272.618224 
iter  10 value 1264.071379
iter  20 value 1088.386618
iter  30 value 1068.588695
iter  40 value 1068.057651
iter  50 value 1068.044700
iter  50 value 1068.044698
final  value 1068.044698 
converged</code></pre>
<pre class="r"><code>summary(fit)</code></pre>
<pre><code>Call:
nnet::multinom(formula = tumor.stage ~ OTU + Abundance + OTU:Abundance, 
    data = dat.wgs.s)

Coefficients:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    1.0009873            0.05475034               -0.03441501
III   0.3731553            0.03709380               -0.13771028
IV   -1.1872774            0.07070086               -0.06956073
    OTUPrevotella melaninogenica    Abundance OTUStreptococcus spp.:Abundance
II                  -0.098054710 0.0001629299                   -1.107990e-03
III                 -0.038182390 0.0002076791                    1.123169e-04
IV                  -0.001567239 0.0002142531                    2.550826e-05
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
II                           0.01520864                           0.0007465488
III                          0.01876677                           0.0006936568
IV                           0.01841968                           0.0006674453

Std. Errors:
    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II   0.04465893            0.03410531               0.003852009
III  0.04935453            0.03859613               0.003523711
IV   0.03407750            0.02646180               0.002556810
    OTUPrevotella melaninogenica    Abundance OTUStreptococcus spp.:Abundance
II                   0.003557919 0.0004797450                    0.0008099566
III                  0.003812940 0.0004792855                    0.0005870014
IV                   0.002684390 0.0004799136                    0.0006106810
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
II                           0.03205088                            0.001302312
III                          0.03204492                            0.001302247
IV                           0.03205537                            0.001305493

Residual Deviance: 2136.089 
AIC: 2184.089 </code></pre>
<pre class="r"><code># obtain significance tests
z &lt;- summary(fit)$coefficients/summary(fit)$standard.errors
z</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II     22.41405             1.6053316                 -8.934302
III     7.56071             0.9610757                -39.081043
IV    -34.84051             2.6718084                -27.206065
    OTUPrevotella melaninogenica Abundance OTUStreptococcus spp.:Abundance
II                   -27.5595665 0.3396177                     -1.36796210
III                  -10.0138986 0.4333098                      0.19134012
IV                    -0.5838345 0.4464409                      0.04177019
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
II                            0.4745154                              0.5732487
III                           0.5856396                              0.5326614
IV                            0.5746208                              0.5112593</code></pre>
<pre class="r"><code>p &lt;- (1 - pnorm(abs(z), 0, 1)) * 2
p</code></pre>
<pre><code>     (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II  0.000000e+00            0.10842086                         0
III 4.019007e-14            0.33651411                         0
IV  0.000000e+00            0.00754437                         0
    OTUPrevotella melaninogenica Abundance OTUStreptococcus spp.:Abundance
II                     0.0000000 0.7341445                       0.1713239
III                    0.0000000 0.6647897                       0.8482591
IV                     0.5593317 0.6552788                       0.9666819
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
II                            0.6351324                              0.5664763
III                           0.5581177                              0.5942680
IV                            0.5655478                              0.6091695</code></pre>
<pre class="r"><code>## extract the coefficients from the model and exponentiate
exp(coef(fit)) # interpret as relative risks</code></pre>
<pre><code>    (Intercept) OTUStreptococcus spp. OTUCampylobacter concisus
II    2.7209669              1.056277                 0.9661705
III   1.4523098              1.037790                 0.8713511
IV    0.3050507              1.073260                 0.9328035
    OTUPrevotella melaninogenica Abundance OTUStreptococcus spp.:Abundance
II                     0.9065993  1.000163                       0.9988926
III                    0.9625374  1.000208                       1.0001123
IV                     0.9984340  1.000214                       1.0000255
    OTUCampylobacter concisus:Abundance OTUPrevotella melaninogenica:Abundance
II                             1.015325                               1.000747
III                            1.018944                               1.000694
IV                             1.018590                               1.000668</code></pre>
<pre class="r"><code># save fitted logits
pp &lt;- fitted(fit)
# predit data
gmeans &lt;- dat.wgs.s %&gt;% group_by(OTU) %&gt;% summarise(M = mean(Abundance))
dotu &lt;- data.frame(OTU = c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), Abundance = gmeans$M)
predict(fit, newdata = dotu, &quot;probs&quot;) # bigger differences</code></pre>
<pre><code>           I        II       III         IV
1 0.15492073 0.5033394 0.2820674 0.05967242
2 0.18245077 0.4709959 0.2851697 0.06138363
3 0.09286417 0.5319037 0.3073122 0.06791994
4 0.06259103 0.5583438 0.3127821 0.06628305</code></pre>
<pre class="r"><code>## look at the averaged predicted probabilities for different values of the continuous predictor variable Abundnace within each level of OTU
dabund &lt;- data.frame(
  OTU = rep(c(&quot;Fusobacterium nucleatum&quot;, &quot;Streptococcus spp.&quot;, &quot;Campylobacter concisus&quot;, &quot;Prevotella melaninogenica&quot;), each = 51),
  Abundance = rep(seq(0, 500,10), 4)
)
pp.abund &lt;-cbind(dabund, predict(fit, newdata = dabund, &quot;probs&quot;, se = TRUE))

## calculate the mean probabilities within each level of OTU
by(pp.abund[, 3:6], pp.abund$OTU, colMeans)</code></pre>
<pre><code>pp.abund$OTU: Campylobacter concisus
         I         II        III         IV 
0.02686404 0.40139010 0.47478711 0.09695875 
------------------------------------------------------------ 
pp.abund$OTU: Fusobacterium nucleatum
         I         II        III         IV 
0.17591465 0.49843817 0.26904262 0.05660455 
------------------------------------------------------------ 
pp.abund$OTU: Prevotella melaninogenica
         I         II        III         IV 
0.16138847 0.49677353 0.28093211 0.06090589 
------------------------------------------------------------ 
pp.abund$OTU: Streptococcus spp.
         I         II        III         IV 
0.18997612 0.43269170 0.31115069 0.06618149 </code></pre>
<pre class="r"><code>## melt data set to long for ggplot2
lpp &lt;- melt(pp.abund, id.vars = c(&quot;OTU&quot;, &quot;Abundance&quot;), value.name = &quot;probability&quot;)

## plot predicted probabilities across Abundance values for each level of OTU
## facetted by tumor.stage
ggplot(lpp, aes(x = Abundance, y = probability, colour = OTU)) +
  geom_line() + 
  facet_grid(variable ~., scales=&quot;free&quot;)+
  labs(y=&quot;Probability of Tumor Stage&quot;,
       title=&quot;Tumor stage likelihood with bacteria abundance and OTU&quot;)+
  theme(
    panel.grid = element_blank()
  )</code></pre>
<p><img src="figure/results-question-3.Rmd/glm-wgs-m3-1.png" width="100%" height="100%" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] cowplot_1.1.1     dendextend_1.14.0 ggdendro_0.1.22   reshape2_1.4.4   
 [5] car_3.0-10        carData_3.0-4     gvlma_1.0.0.3     patchwork_1.1.1  
 [9] viridis_0.5.1     viridisLite_0.3.0 gridExtra_2.3     xtable_1.8-4     
[13] kableExtra_1.3.1  data.table_1.13.6 readxl_1.3.1      forcats_0.5.1    
[17] stringr_1.4.0     dplyr_1.0.3       purrr_0.3.4       readr_1.4.0      
[21] tidyr_1.1.2       tibble_3.0.6      ggplot2_3.3.3     tidyverse_1.3.0  
[25] lmerTest_3.1-3    lme4_1.1-26       Matrix_1.2-18     vegan_2.5-7      
[29] lattice_0.20-41   permute_0.9-5     phyloseq_1.34.0   workflowr_1.6.2  

loaded via a namespace (and not attached):
  [1] minqa_1.2.4         colorspace_2.0-0    rio_0.5.16         
  [4] ellipsis_0.3.1      rprojroot_2.0.2     XVector_0.30.0     
  [7] fs_1.5.0            rstudioapi_0.13     farver_2.0.3       
 [10] lubridate_1.7.9.2   xml2_1.3.2          codetools_0.2-16   
 [13] splines_4.0.3       knitr_1.31          ade4_1.7-16        
 [16] jsonlite_1.7.2      nloptr_1.2.2.2      broom_0.7.4        
 [19] cluster_2.1.0       dbplyr_2.1.0        BiocManager_1.30.10
 [22] compiler_4.0.3      httr_1.4.2          backports_1.2.1    
 [25] assertthat_0.2.1    cli_2.3.0           later_1.1.0.1      
 [28] htmltools_0.5.1.1   prettyunits_1.1.1   tools_4.0.3        
 [31] igraph_1.2.6        gtable_0.3.0        glue_1.4.2         
 [34] Rcpp_1.0.6          Biobase_2.50.0      cellranger_1.1.0   
 [37] vctrs_0.3.6         Biostrings_2.58.0   rhdf5filters_1.2.0 
 [40] multtest_2.46.0     ape_5.4-1           nlme_3.1-149       
 [43] iterators_1.0.13    xfun_0.20           ps_1.5.0           
 [46] openxlsx_4.2.3      rvest_0.3.6         lifecycle_0.2.0    
 [49] statmod_1.4.35      zlibbioc_1.36.0     MASS_7.3-53        
 [52] scales_1.1.1        hms_1.0.0           promises_1.1.1     
 [55] parallel_4.0.3      biomformat_1.18.0   rhdf5_2.34.0       
 [58] curl_4.3            yaml_2.2.1          stringi_1.5.3      
 [61] highr_0.8           S4Vectors_0.28.1    foreach_1.5.1      
 [64] BiocGenerics_0.36.0 zip_2.1.1           boot_1.3-25        
 [67] rlang_0.4.10        pkgconfig_2.0.3     evaluate_0.14      
 [70] Rhdf5lib_1.12.1     labeling_0.4.2      tidyselect_1.1.0   
 [73] plyr_1.8.6          magrittr_2.0.1      R6_2.5.0           
 [76] IRanges_2.24.1      generics_0.1.0      DBI_1.1.1          
 [79] foreign_0.8-80      pillar_1.4.7        haven_2.3.1        
 [82] whisker_0.4         withr_2.4.1         mgcv_1.8-33        
 [85] nnet_7.3-14         abind_1.4-5         survival_3.2-7     
 [88] modelr_0.1.8        crayon_1.4.1        rmarkdown_2.6      
 [91] progress_1.2.2      grid_4.0.3          git2r_0.28.0       
 [94] reprex_1.0.0        digest_0.6.27       webshot_0.5.2      
 [97] httpuv_1.5.5        numDeriv_2016.8-1.1 stats4_4.0.3       
[100] munsell_0.5.0      </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
