<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Results Output for Question 1</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">esoph-micro-cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Results Output for Question 1</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>esoph-micro-cancer-workflow/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200916code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200916)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200916code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200916)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomGreathouseLabesophmicrocancerworkflowtree8f2ce3f3d7186cf9fda007822dd99154611a1384targetblank8f2ce3fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/tree/8f2ce3f3d7186cf9fda007822dd99154611a1384" target="_blank">8f2ce3f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomGreathouseLabesophmicrocancerworkflowtree8f2ce3f3d7186cf9fda007822dd99154611a1384targetblank8f2ce3fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/tree/8f2ce3f3d7186cf9fda007822dd99154611a1384" target="_blank">8f2ce3f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/

Untracked files:
    Untracked:  analysis/slide-figures-heatmaps-dendrograms-updates.Rmd
    Untracked:  output/New folder/
    Untracked:  output/slide-5-heatmap-001.pdf
    Untracked:  output/slide-5-heatmap-001.png
    Untracked:  output/slide-6-heatmap-001.pdf
    Untracked:  output/slide-6-heatmap-001.png
    Untracked:  output/slide-7-heatmap-001.pdf
    Untracked:  output/slide-7-heatmap-001.png

Unstaged changes:
    Modified:   analysis/index.Rmd
    Modified:   analysis/results-question-1.Rmd
    Modified:   analysis/results-question-3.Rmd
    Modified:   code/heatmap-dendrogram-slide-5-05.R
    Modified:   code/load_packages.R
    Modified:   output/slide-10-heatmap.pdf
    Modified:   output/slide-10-heatmap.png
    Modified:   output/slide-5-heatmap-01.pdf
    Modified:   output/slide-5-heatmap-01.png
    Modified:   output/slide-5-heatmap-05.pdf
    Modified:   output/slide-5-heatmap-05.png
    Modified:   output/slide-6-heatmap-01.pdf
    Modified:   output/slide-6-heatmap-01.png
    Modified:   output/slide-6-heatmap-05.pdf
    Modified:   output/slide-6-heatmap-05.png
    Modified:   output/slide-7-heatmap-01.pdf
    Modified:   output/slide-7-heatmap-01.png
    Modified:   output/slide-7-heatmap-05.pdf
    Modified:   output/slide-7-heatmap-05.png
    Modified:   output/slide-8-heatmap.pdf
    Modified:   output/slide-8-heatmap.png
    Modified:   output/slide-9-heatmap.pdf
    Modified:   output/slide-9-heatmap.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/results-question-1.Rmd</code>) and HTML (<code>docs/results-question-1.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/blob/65859077bc3495e2e48531d010b0594ebf4089f4/analysis/results-question-1.Rmd" target="_blank">6585907</a>
</td>
<td>
noah-padgett
</td>
<td>
2021-01-21
</td>
<td>
updated figures with dendrogram
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/GreathouseLab/esoph-micro-cancer-workflow/a79cf3a77f6ddebefc0edd18a8117819c61c50b4/docs/results-question-1.html" target="_blank">a79cf3a</a>
</td>
<td>
noah-padgett
</td>
<td>
2021-01-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/blob/e6b4c7f6c88be3a960c73d2004797f323eb093b4/analysis/results-question-1.Rmd" target="_blank">e6b4c7f</a>
</td>
<td>
noah-padgett
</td>
<td>
2021-01-14
</td>
<td>
meeting update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/GreathouseLab/esoph-micro-cancer-workflow/e6b4c7f6c88be3a960c73d2004797f323eb093b4/docs/results-question-1.html" target="_blank">e6b4c7f</a>
</td>
<td>
noah-padgett
</td>
<td>
2021-01-14
</td>
<td>
meeting update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="question-1" class="section level1">
<h1>Question 1</h1>
<pre><code>Q1: is there a taxonomic signature shared between the barrett&#39;s samples?</code></pre>
<ul>
<li>Heatmap of relative abundance supervised by sample type: Barrett’s (BO), tumor-adjacent EAC-w/history of barrett’s, EAC-w/ history of barrett’s</li>
<li>Stacked bar chart of phylum and genus abundance by sample type, same as above</li>
<li>Additional comparison in TCGA; EAC w/history of Barrett’s vs EAC w/ no history of Barrett’s; same analyses as above</li>
</ul>
<p><strong>FILE HAS BEEN DEPRICATED</strong></p>
</div>
<div id="summary-of-observations" class="section level1">
<h1>Summary of observations</h1>
<div id="nci-16s-data" class="section level2">
<h2>NCI 16s data</h2>
<pre class="r"><code># in long format
table(dat.16s$sample_type)</code></pre>
<pre><code>
                                      0                           Barretts Only 
                                  19800                                    1320 
EAC-adjacent tissue w/ Barretts History         EAC tissues w/ Barretts History 
                                  11352                                    9240 </code></pre>
<pre class="r"><code>dat &lt;- dat.16s %&gt;% filter(OTU == &quot;Fusobacterium_nucleatum&quot;)
table(dat$sample_type)</code></pre>
<pre><code>
                                      0                           Barretts Only 
                                     75                                       5 
EAC-adjacent tissue w/ Barretts History         EAC tissues w/ Barretts History 
                                     43                                      35 </code></pre>
<pre class="r"><code>table(dat$Barretts.)</code></pre>
<pre><code>
 N  Y 
71 87 </code></pre>
<p>5 Barretts samples that were non-EAC tissue related.</p>
</div>
<div id="tcga-rnaseq-data" class="section level2">
<h2>TCGA RNAseq data</h2>
<pre class="r"><code># in long format
table(dat.rna$sample_type)</code></pre>
<pre><code>
                                      0 EAC-adjacent tissue w/ Barretts History 
                                 112176                                    2337 
        EAC tissues w/ Barretts History 
                                  20254 </code></pre>
<pre class="r"><code>dat &lt;- dat.rna %&gt;% filter(otu2 == &quot;Fusobacterium nucleatum&quot;)
table(dat$sample_type)</code></pre>
<pre><code>
                                      0 EAC-adjacent tissue w/ Barretts History 
                                    144                                       3 
        EAC tissues w/ Barretts History 
                                     26 </code></pre>
<pre class="r"><code>table(dat$Barrett.s.Esophagus.Reported)</code></pre>
<pre><code>
           No Not Available           Yes 
          113            31            29 </code></pre>
</div>
<div id="tcga-wgs-data" class="section level2">
<h2>TCGA WGS data</h2>
<pre class="r"><code># in long format
table(dat.wgs$sample_type)</code></pre>
<pre><code>
                                      0 EAC-adjacent tissue w/ Barretts History 
                                 100491                                    4674 
        EAC tissues w/ Barretts History 
                                   3116 </code></pre>
<pre class="r"><code>dat &lt;- dat.wgs %&gt;% filter(otu2 == &quot;Fusobacterium nucleatum&quot;)
table(dat$sample_type)</code></pre>
<pre><code>
                                      0 EAC-adjacent tissue w/ Barretts History 
                                    129                                       6 
        EAC tissues w/ Barretts History 
                                      4 </code></pre>
<pre class="r"><code>table(dat$Barrett.s.Esophagus.Reported)</code></pre>
<pre><code>
           No Not Available           Yes 
           54            47            10 </code></pre>
<!-- # Heatmaps -->
<!-- 5% RA >= per individual -->
<!-- ## All OTUs (RA > X) -->
<!-- ```{r heatmap-nci-genus-1, out.height="300%", out.width="300%"} -->
<!-- plot.dat <- dat.16s %>% filter(sample_type != "0", Abundance > 0.01) %>% -->
<!--   mutate(ID = as.factor(accession.number), -->
<!--          Genus = substr(Genus, 4, 1000), -->
<!--          Phylum = substr(Phylum, 4, 1000)) %>% -->
<!--   select(sample_type, Phylum, Genus, ID, Abundance) -->
<!-- plot.dat <- na.omit(plot.dat)  -->
<!-- a <- plot.dat %>% -->
<!--   expand(ID, sample_type, Genus) -->
<!-- a$Abundance <- 0 -->
<!-- plot.dat <- full_join(plot.dat, a) -->
<!-- # For the full dendrogram -->
<!-- library(plyr) -->
<!-- library(reshape2) -->
<!-- library(dplyr) -->
<!-- library(ggplot2) -->
<!-- library(ggdendro) -->
<!-- library(gridExtra) -->
<!-- library(dendextend) -->
<!-- set.seed(10) -->
<!-- # The source data -->
<!-- mat <- matrix(rnorm(24 * 10, mean = 1, sd = 2),  -->
<!--               nrow = 24, ncol = 10,  -->
<!--               dimnames = list(paste("g", 1:24, sep = ""),  -->
<!--                               paste("sample", 1:10, sep = ""))) -->
<!-- sample_names <- colnames(mat) -->
<!-- # Obtain the dendrogram -->
<!-- dend <- as.dendrogram(hclust(dist(mat))) -->
<!-- dend_data <- dendro_data(dend) -->
<!-- # Setup the data, so that the layout is inverted (this is more  -->
<!-- # "clear" than simply using coord_flip()) -->
<!-- segment_data <- with( -->
<!--     segment(dend_data),  -->
<!--     data.frame(x = y, y = x, xend = yend, yend = xend)) -->
<!-- # Use the dendrogram label data to position the gene labels -->
<!-- gene_pos_table <- with( -->
<!--     dend_data$labels,  -->
<!--     data.frame(y_center = x, gene = as.character(label), height = 1)) -->
<!-- # Table to position the samples -->
<!-- sample_pos_table <- data.frame(sample = sample_names) %>% -->
<!--     dplyr::mutate(x_center = (1:n()), width = 1) -->
<!-- # Neglecting the gap parameters -->
<!-- heatmap_data <- mat %>%  -->
<!--     reshape2::melt(value.name = "expr", varnames = c("gene", "sample")) %>% -->
<!--     left_join(gene_pos_table) %>% -->
<!--     left_join(sample_pos_table) -->
<!-- # Limits for the vertical axes -->
<!-- gene_axis_limits <- with( -->
<!--     gene_pos_table,  -->
<!--     c(min(y_center - 0.5 * height), max(y_center + 0.5 * height)) -->
<!-- ) +  -->
<!--     0.1 * c(-1, 1) # extra spacing: 0.1 -->
<!-- # Heatmap plot -->
<!-- plt_hmap <- ggplot(heatmap_data,  -->
<!--                    aes(x = x_center, y = y_center, fill = expr,  -->
<!--                        height = height, width = width)) +  -->
<!--     geom_tile() + -->
<!--     scale_fill_gradient2("expr", high = "darkred", low = "darkblue") + -->
<!--     scale_x_continuous(breaks = sample_pos_table$x_center,  -->
<!--                        labels = sample_pos_table$sample,  -->
<!--                        expand = c(0, 0)) +  -->
<!--     # For the y axis, alternatively set the labels as: gene_position_table$gene -->
<!--     scale_y_continuous(breaks = gene_pos_table[, "y_center"],  -->
<!--                        labels = rep("", nrow(gene_pos_table)), -->
<!--                        limits = gene_axis_limits,  -->
<!--                        expand = c(0, 0)) +  -->
<!--     labs(x = "Sample", y = "") + -->
<!--     theme_bw() + -->
<!--     theme(axis.text.x = element_text(size = rel(1), hjust = 1, angle = 45),  -->
<!--           # margin: top, right, bottom, and left -->
<!--           plot.margin = unit(c(1, 0.2, 0.2, -0.7), "cm"),  -->
<!--           panel.grid.minor = element_blank()) -->
<!-- # Dendrogram plot -->
<!-- plt_dendr <- ggplot(segment_data) +  -->
<!--     geom_segment(aes(x = x, y = y, xend = xend, yend = yend)) +  -->
<!--     scale_x_reverse(expand = c(0, 0.5)) +  -->
<!--     scale_y_continuous(breaks = gene_pos_table$y_center,  -->
<!--                        labels = gene_pos_table$gene,  -->
<!--                        limits = gene_axis_limits,  -->
<!--                        expand = c(0, 0)) +  -->
<!--     labs(x = "Distance", y = "", colour = "", size = "") + -->
<!--     theme_bw() +  -->
<!--     theme(panel.grid.minor = element_blank()) -->
<!-- library(cowplot) -->
<!-- plot_grid(plt_dendr, plt_hmap, align = 'h', rel_widths = c(1, 1)) -->
<!-- # dendrogram -->
<!-- library("ggdendro") -->
<!-- # widen plot.dat for dendro -->
<!-- dat.wide <- plot.dat %>% -->
<!--   mutate( -->
<!--     ID = paste0(ID, "_",sample_type) -->
<!--   ) %>% -->
<!--   select(ID, Genus, Abundance) %>% -->
<!--   pivot_wider( -->
<!--     id_cols = Genus, -->
<!--     names_from = ID, -->
<!--     values_from = Abundance, -->
<!--     values_fill = 0 -->
<!--   ) -->
<!-- rn <- dat.wide$Genus -->
<!-- dat.wide <- as.matrix(dat.wide[,-1]) -->
<!-- rownames(dat.wide) <- rn -->
<!-- hc <- hclust(dist(dat.wide), "ave") -->
<!-- heatmaply(dat.wide,) -->
<!-- p <- ggdendrogram(hc, rotate = TRUE) -->
<!-- p <- p + theme(axis.text.x = element_blank(), -->
<!--                axis.text.y = element_blank()) -->
<!-- p -->
<!-- # extract genus order -->
<!-- bugorder <- hc$labels[hc$order] -->
<!-- plot.dat$Genus <- factor( -->
<!--   plot.dat$Genus, -->
<!--   levels=bugorder, -->
<!--   ordered=T) -->
<!-- p1 <- ggplot(plot.dat, aes(x = ID, y = Genus, fill = Abundance)) + -->
<!--   geom_tile()+ -->
<!--   labs(title="NCI 16s Data", subtitle = "Genus Level Data", -->
<!--        x = "ID/Accession Number") + -->
<!--   facet_grid(.~sample_type, scales="free")+ -->
<!--   scale_fill_gradient(low="white", high="black", trans="sqrt", limits=c(0, 0.5)) + -->
<!--   theme_classic()+ -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5), -->
<!--     strip.text.y = element_text(angle = 0), -->
<!--     legend.position = "bottom" -->
<!--   ) -->
<!-- p1 + p + plot_layout(guides='collect') & -->
<!--   theme(legend.position='bottom') -->
<!-- #ggsave("data/heatmap_nci16s_genus_slide5.pdf", p1, units="in", height=23, width=16) -->
<!-- ``` -->
<!-- ```{r heatmap-nci-genus-2, out.height="300%", out.width="300%"} -->
<!-- plot.dat <- dat.16s %>% filter(sample_type != "0", Abundance > 0.05) %>% -->
<!--   mutate(ID = as.factor(accession.number), -->
<!--          Genus = substr(Genus, 4, 1000), -->
<!--          Phylum = substr(Phylum, 4, 1000)) %>% -->
<!--   select(sample_type, Phylum, Genus, ID, Abundance) -->
<!-- plot.dat <- na.omit(plot.dat)  -->
<!-- p1 <- ggplot(plot.dat, aes(x = ID, y = Genus, fill = Abundance)) + -->
<!--   geom_tile()+ -->
<!--   labs(title="NCI 16s Data", -->
<!--        subtitle = "Genus Level Data", -->
<!--        x = "ID/Accession Number") + -->
<!--   facet_grid(.~sample_type, scales="free")+ -->
<!--   scale_fill_gradient(low="white", high="black", trans="sqrt", limits=c(0, 0.5)) + -->
<!--   theme_classic()+ -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5), -->
<!--     strip.text.y = element_text(angle = 0) -->
<!--   ) -->
<!-- p1 -->
<!-- ggsave("data/heatmap_nci16s_genus_slide6.pdf", p1, units="in", height=23, width=16) -->
<!-- ``` -->
<!-- ```{r heatmap-nci-phylum, out.height="1000%", out.width="400%"} -->
<!-- plot.dat <- dat.16s %>% filter(sample_type != "0") %>% -->
<!--   mutate(ID = as.factor(accession.number), -->
<!--          Genus = substr(Genus, 4, 1000), -->
<!--          Phylum = substr(Phylum, 4, 1000)) %>% -->
<!--   select(sample_type, Phylum, Genus, ID, Abundance) -->
<!-- plot.dat <- na.omit(plot.dat)  -->
<!-- p1 <- ggplot(plot.dat, aes(x = ID, y = Phylum, fill = Abundance)) + -->
<!--   geom_tile()+ -->
<!--   labs(title="NCI 16s Data", subtitle = "Phylum Level Data", -->
<!--        x = "ID/Accession Number") + -->
<!--   facet_grid(.~sample_type, scales="free")+ -->
<!--   scale_fill_gradient(low="white", high="black", trans="sqrt", limits=c(0, 0.5)) + -->
<!--   theme_classic()+ -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5), -->
<!--     strip.text.y = element_text(angle = 0) -->
<!--   ) -->
<!-- p1 -->
<!-- #ggsave("data/heatmap_nci16s_phylum.pdf", p1, units="in", height=10, width=16) -->
<!-- ``` -->
<!-- ```{r heatmap-tcga-rna-genus, out.height="300%", out.width="300%"} -->
<!-- plot.dat <- dat.rna %>% filter(sample_type != "0", Abundance > 0.05) %>% -->
<!--   select(sample_type, Phylum, Genus, Patient_ID, Abundance) -->
<!-- plot.dat <- na.omit(plot.dat)  -->
<!-- p1 <- ggplot(plot.dat, aes(x = Patient_ID, y = Genus, fill = Abundance)) + -->
<!--   geom_tile()+ -->
<!--   labs(title="TCGA RNAseq Data", subtitle = "Genus Level Data", -->
<!--        x = "ID/Accession Number") + -->
<!--   facet_grid(.~sample_type, scales="free")+ -->
<!--   scale_fill_gradient(low="white", high="black", trans="sqrt", limits=c(0, 0.5)) + -->
<!--   theme_classic()+ -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5), -->
<!--     strip.text.y = element_text(angle = 0) -->
<!--   ) -->
<!-- p1 -->
<!-- #ggsave("data/heatmap_tcgarna_genus.pdf", p1, units="in", height=23, width=16) -->
<!-- ``` -->
<!-- ```{r heatmap-tcga-rna-phylum, out.height="1000%", out.width="400%"} -->
<!-- plot.dat <- dat.rna %>% filter(sample_type != "0") %>% -->
<!--   select(sample_type, Phylum, Genus, Patient_ID, Abundance) -->
<!-- plot.dat <- na.omit(plot.dat)  -->
<!-- p1 <- ggplot(plot.dat, aes(x = Patient_ID, y = Phylum, fill = Abundance)) + -->
<!--   geom_tile()+ -->
<!--   labs(title="NCI 16s Data", subtitle = "Phylum Level Data", -->
<!--        x = "ID/Accession Number") + -->
<!--   facet_grid(.~sample_type, scales="free")+ -->
<!--   scale_fill_gradient(low="white", high="black", trans="sqrt", limits=c(0, 0.5)) + -->
<!--   theme_classic()+ -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5), -->
<!--     strip.text.y = element_text(angle = 0) -->
<!--   ) -->
<!-- p1 -->
<!-- ``` -->
<!-- ```{r heatmap-tcga-wgs-genus, out.height="300%", out.width="300%"} -->
<!-- plot.dat <- dat.wgs %>% filter(sample_type != "0", Abundance > 0.001) %>% -->
<!--   select(sample_type, Phylum, Genus, Patient_ID, Abundance) -->
<!-- plot.dat <- na.omit(plot.dat)  -->
<!-- p1 <- ggplot(plot.dat, aes(x = Patient_ID, y = Genus, fill = Abundance)) + -->
<!--   geom_tile()+ -->
<!--   labs(title="NCI 16s Data", subtitle = "Genus Level Data", -->
<!--        x = "ID/Accession Number") + -->
<!--   facet_grid(.~sample_type, scales="free")+ -->
<!--   scale_fill_gradient(low="white", high="black", trans="sqrt", limits=c(0, 0.5)) + -->
<!--   theme_classic()+ -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5), -->
<!--     strip.text.y = element_text(angle = 0) -->
<!--   ) -->
<!-- p1 -->
<!-- #ggsave("data/heatmap_nci16s_genus.pdf", p1, units="in", height=23, width=16) -->
<!-- ``` -->
<!-- ```{r heatmap-tcga-wgs-phylum, out.height="1000%", out.width="400%"} -->
<!-- plot.dat <- dat.wgs %>% filter(sample_type != "0") %>% -->
<!--   select(sample_type, Phylum, Genus, Patient_ID, Abundance) -->
<!-- plot.dat <- na.omit(plot.dat)  -->
<!-- p1 <- ggplot(plot.dat, aes(x = Patient_ID, y = Phylum, fill = Abundance)) + -->
<!--   geom_tile()+ -->
<!--   labs(title="NCI 16s Data", subtitle = "Phylum Level Data", -->
<!--        x = "ID/Accession Number") + -->
<!--   facet_grid(.~sample_type, scales="free")+ -->
<!--   scale_fill_gradient(low="white", high="black", trans="sqrt", limits=c(0, 0.5)) + -->
<!--   theme_classic()+ -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5), -->
<!--     strip.text.y = element_text(angle = 0) -->
<!--   ) -->
<!-- p1 -->
<!-- ``` -->
<!-- ## Specific OTUs -->
<!-- ```{r heatmap-nci-otu, out.height="300%", out.width="300%"} -->
<!-- plot.dat <- dat.16s.s %>% filter(sample_type != "0") %>% -->
<!--   mutate(ID = as.factor(accession.number), -->
<!--          Genus = substr(Genus, 4, 1000), -->
<!--          Phylum = substr(Phylum, 4, 1000)) %>% -->
<!--   select(sample_type, Phylum, Genus, OTU, ID, Abundance) -->
<!-- plot.dat <- na.omit(plot.dat)  -->
<!-- p1 <- ggplot(plot.dat, aes(x = ID, y = OTU, fill = Abundance)) + -->
<!--   geom_tile()+ -->
<!--   labs(title="NCI 16s Data", y=NULL, -->
<!--        x = "ID/Accession Number") + -->
<!--   facet_grid(.~sample_type, scales="free")+ -->
<!--   scale_fill_gradient(low="white", high="black", trans="sqrt", limits=c(0, 0.5)) + -->
<!--   theme_classic()+ -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5), -->
<!--     strip.text.y = element_text(angle = 0) -->
<!--   ) -->
<!-- p1 -->
<!-- ``` -->
<!-- ```{r heatmap-tcgarna, out.height="300%", out.width="300%"} -->
<!-- plot.dat <- dat.rna.s %>% filter(sample_type != "0", is.na(OTU1) == F) %>% -->
<!--   select(sample_type, Phylum, Genus, OTU, Patient_ID, Abundance) -->
<!-- plot.dat <- na.omit(plot.dat)  -->
<!-- p1 <- ggplot(plot.dat, aes(x = Patient_ID, y = OTU, fill = Abundance)) + -->
<!--   geom_tile()+ -->
<!--   labs(title="TCGA RNAseq Data", y=NULL, -->
<!--        x = "ID/Accession Number") + -->
<!--   facet_grid(.~sample_type, scales="free")+ -->
<!--   scale_fill_gradient(low="white", high="black", trans="sqrt", limits=c(0, 0.5)) + -->
<!--   theme_classic()+ -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5), -->
<!--     strip.text.y = element_text(angle = 0) -->
<!--   ) -->
<!-- p1 -->
<!-- ``` -->
<!-- ```{r heatmap-tcgawgs, out.height="100%", out.width="400%"} -->
<!-- plot.dat <- dat.wgs.s %>% filter(sample_type != "0", is.na(OTU1) == F) %>% -->
<!--   select(sample_type, Phylum, Genus, OTU, Patient_ID, Abundance) -->
<!-- plot.dat <- na.omit(plot.dat)  -->
<!-- p1 <- ggplot(plot.dat, aes(x = Patient_ID, y = OTU, fill = Abundance)) + -->
<!--   geom_tile()+ -->
<!--   labs(title="TCGA WGS Data", y=NULL, -->
<!--        x = "ID/Accession Number") + -->
<!--   facet_grid(.~sample_type, scales="free")+ -->
<!--   scale_fill_gradient(low="white", high="black", trans="sqrt", limits=c(0, 0.5)) + -->
<!--   theme_classic()+ -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5), -->
<!--     strip.text.y = element_text(angle = 0) -->
<!--   ) -->
<!-- p1 -->
<!-- ``` -->
<!-- # Stacked Bar Charts -->
<!-- ## All OTUs (RA > 0.001) -->
<!-- ```{r bar-nci-genus, out.height="300%", out.width="300%"} -->
<!-- plot.dat <- dat.16s %>% filter(sample_type != "0", Abundance > 0.01 ) %>% -->
<!--   mutate(ID = as.factor(accession.number), -->
<!--          Genus = substr(Genus, 4, 1000), -->
<!--          Phylum = substr(Phylum, 4, 1000))%>% -->
<!--   dplyr::group_by(sample_type, Genus)%>% -->
<!--   dplyr::summarise( -->
<!--     Abundance = mean(Abundance, na.rm=T) -->
<!--   ) -->
<!-- p1 <- ggplot(plot.dat, aes(x=sample_type, y = Abundance, fill=Genus)) + -->
<!--   geom_bar(stat="identity")+ -->
<!--   labs(title="NCI 16s Data", -->
<!--        subtitle = "Genus Level Data", -->
<!--        x = "Tissue Group", -->
<!--        y="Average Relative Abundance") + -->
<!--   theme_classic()+ -->
<!--   theme(legend.position = "none") -->
<!-- p1 -->
<!-- ``` -->
<!-- TOO MANY LEVELS FOR IT TO MAKE SENSE -->
<!-- ```{r bar-nci-phylum, out.height="1000%", out.width="400%"} -->
<!-- plot.dat <- dat.16s %>% filter(sample_type != "0") %>% -->
<!--   mutate(ID = as.factor(accession.number), -->
<!--          Genus = substr(Genus, 4, 1000), -->
<!--          Phylum = substr(Phylum, 4, 1000))%>% -->
<!--   dplyr::group_by(sample_type, Phylum)%>% -->
<!--   dplyr::summarise( -->
<!--     Abundance = mean(Abundance, na.rm=T) -->
<!--   ) -->
<!-- p1 <- ggplot(plot.dat, aes(x=sample_type, y = Abundance, fill=Phylum)) + -->
<!--   geom_bar(stat="identity")+ -->
<!--   labs(title="NCI 16s Data", -->
<!--        subtitle = "Phylum Level Data", -->
<!--        x = "Tissue Group", -->
<!--        y="Average Relative Abundance") + -->
<!--   theme_classic() -->
<!-- p1  -->
<!-- ``` -->
<!-- ```{r bar-tcga-rna-genus, out.height="300%", out.width="300%"} -->
<!-- plot.dat <- dat.rna %>% filter(sample_type != "0", Abundance > 0.001)%>% -->
<!--   dplyr::group_by(sample_type, Genus)%>% -->
<!--   dplyr::summarise( -->
<!--     Abundance = mean(Abundance, na.rm=T) -->
<!--   ) -->
<!-- p1 <- ggplot(plot.dat, aes(x=sample_type, y = Abundance, fill=Genus)) + -->
<!--   geom_bar(stat="identity")+ -->
<!--   labs(title="TCGA RNAseq Data", -->
<!--        subtitle = "Genus Level Data", -->
<!--        x = "Tissue Group", -->
<!--        y="Average Relative Abundance") + -->
<!--   theme_classic() -->
<!-- p1 -->
<!-- #ggsave("data/bar_tcgarna_genus.pdf", p1, units="in", height=23, width=16) -->
<!-- ``` -->
<!-- ```{r bar-tcga-rna-phylum, out.height="1000%", out.width="400%"} -->
<!-- plot.dat <- dat.rna %>% filter(sample_type != "0")%>% -->
<!--   dplyr::group_by(sample_type, Phylum)%>% -->
<!--   dplyr::summarise( -->
<!--     Abundance = mean(Abundance, na.rm=T) -->
<!--   ) -->
<!-- p1 <- ggplot(plot.dat, aes(x=sample_type, y = Abundance, fill=Phylum)) + -->
<!--   geom_bar(stat="identity")+ -->
<!--   labs(title="TCGA RNAseq Data", -->
<!--        subtitle = "Phylum Level Data", -->
<!--        x = "Tissue Group", -->
<!--        y="Average Relative Abundance") + -->
<!--   theme_classic() -->
<!-- p1 -->
<!-- ``` -->
<!-- ```{r bar-tcga-wgs-genus, out.height="300%", out.width="300%"} -->
<!-- plot.dat <- dat.wgs %>% filter(sample_type != "0", Abundance > 0.001)%>% -->
<!--   dplyr::group_by(sample_type, Genus)%>% -->
<!--   dplyr::summarise( -->
<!--     Abundance = mean(Abundance, na.rm=T) -->
<!--   ) -->
<!-- p1 <- ggplot(plot.dat, aes(x=sample_type, y = Abundance, fill=Genus)) + -->
<!--   geom_bar(stat="identity")+ -->
<!--   labs(title="TCGA WGS Data", -->
<!--        subtitle = "Genus Level Data", -->
<!--        x = "Tissue Group", -->
<!--        y="Average Relative Abundance") + -->
<!--   theme_classic() -->
<!-- p1 -->
<!-- ``` -->
<!-- ```{r bar-tcga-wgs-phylum, out.height="1000%", out.width="400%"} -->
<!-- plot.dat <- dat.wgs %>% filter(sample_type != "0")%>% -->
<!--   dplyr::group_by(sample_type, Phylum)%>% -->
<!--   dplyr::summarise( -->
<!--     Abundance = mean(Abundance, na.rm=T) -->
<!--   ) -->
<!-- p1 <- ggplot(plot.dat, aes(x=sample_type, y = Abundance, fill=Phylum)) + -->
<!--   geom_bar(stat="identity")+ -->
<!--   labs(title="TCGA WGS Data", -->
<!--        subtitle = "Phylum Level Data", -->
<!--        x = "Tissue Group", -->
<!--        y="Average Relative Abundance") + -->
<!--   theme_classic() -->
<!-- p1 -->
<!-- ``` -->
<!-- ## Specific OTUs -->
<!-- ```{r bar-nci-otu, out.height="300%", out.width="300%"} -->
<!-- plot.dat <- dat.16s.s %>% filter(sample_type != "0") %>% -->
<!--   mutate(ID = as.factor(accession.number), -->
<!--          Genus = substr(Genus, 4, 1000), -->
<!--          Phylum = substr(Phylum, 4, 1000))%>% -->
<!--   dplyr::group_by(sample_type, OTU1)%>% -->
<!--   dplyr::summarise( -->
<!--     Abundance = mean(Abundance, na.rm=T) -->
<!--   ) -->
<!-- p1 <- ggplot(plot.dat, aes(x=sample_type, y = Abundance, fill=OTU1)) + -->
<!--   geom_bar(stat="identity")+ -->
<!--   labs(title="NCI 16s Data", -->
<!--        x = "Tissue Group", -->
<!--        y="Average Relative Abundance") + -->
<!--   theme_classic()+ -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle=30, vjust=0.95, hjust=0.9), -->
<!--     legend.title = element_blank() -->
<!--   ) -->
<!-- p1 -->
<!-- #ggsave("output/slide-11-bar.pdf", plot=p1, units="in", width=8, height=5) -->
<!-- #ggsave("output/slide-11-bar.png", plot=p1, units="in", width=8, height=5) -->
<!-- ``` -->
<!-- ```{r bar-tcgarna, out.height="300%", out.width="300%"} -->
<!-- plot.dat <- dat.rna.s %>% filter(sample_type != "0", is.na(OTU1) == F)%>% -->
<!--   dplyr::group_by(sample_type, OTU1)%>% -->
<!--   dplyr::summarise( -->
<!--     Abundance = mean(Abundance, na.rm=T) -->
<!--   ) -->
<!-- p1 <- ggplot(plot.dat, aes(x=sample_type, y = Abundance, fill=OTU1)) + -->
<!--   geom_bar(stat="identity")+ -->
<!--   labs(title="TCGA RNAseq Data", -->
<!--        x = "Tissue Group", -->
<!--        y="Average Relative Abundance") + -->
<!--   theme_classic() + -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle=30, vjust=0.95, hjust=0.9), -->
<!--     legend.title = element_blank() -->
<!--   ) -->
<!-- p1 -->
<!-- #ggsave("output/slide-12-bar.pdf", plot=p1, units="in", width=8, height=5) -->
<!-- #ggsave("output/slide-12-bar.png", plot=p1, units="in", width=8, height=5) -->
<!-- ``` -->
<!-- ```{r bar-tcgawgs, out.height="100%", out.width="400%"} -->
<!-- plot.dat <- dat.wgs.s %>% filter(sample_type != "0", is.na(OTU1) == F)%>% -->
<!--   dplyr::group_by(sample_type, OTU1)%>% -->
<!--   dplyr::summarise( -->
<!--     Abundance = mean(Abundance, na.rm=T) -->
<!--   ) -->
<!-- p1 <- ggplot(plot.dat, aes(x=sample_type, y = Abundance, fill=OTU1)) + -->
<!--   geom_bar(stat="identity")+ -->
<!--   labs(title="TCGA WGS Data", -->
<!--        x = "Tissue Group", -->
<!--        y="Average Relative Abundance") + -->
<!--   scale_y_continuous(trans="sqrt", breaks=c(0,0.01, 0.1, 0.2, 0.4, 0.6))+ -->
<!--   theme_classic() + -->
<!--   theme( -->
<!--     axis.text.x = element_text(angle=30, vjust=0.95, hjust=0.9), -->
<!--     legend.title = element_blank() -->
<!--   ) -->
<!-- p1 -->
<!-- #ggsave("output/slide-13-bar.pdf", plot=p1, units="in", width=8, height=5) -->
<!-- #ggsave("output/slide-13-bar.png", plot=p1, units="in", width=8, height=5) -->
<!-- ``` -->
</div>
</div>
<div id="eac-barretts-to-no-barretts-comparison" class="section level1">
<h1>EAC Barretts to No Barretts Comparison</h1>
<div id="heatmaps" class="section level2">
<h2>Heatmaps</h2>
<div id="all-otus-ra-0.001" class="section level3">
<h3>All OTUs (RA &gt; 0.001)</h3>
<pre class="r"><code>plot.dat &lt;- dat.rna %&gt;% filter(EACcomp != &quot;0&quot;, Abundance &gt; 0.001)

p1 &lt;- ggplot(plot.dat, aes(x = Patient_ID, y = Genus, fill = Abundance)) +
  geom_tile()+
  labs(title=&quot;TCGA RNAseq Data&quot;, subtitle = &quot;Genus Level Data&quot;,
       x = &quot;ID/Accession Number&quot;) +
  facet_grid(.~EACcomp, scales=&quot;free&quot;)+
  scale_fill_gradient(low=&quot;white&quot;, high=&quot;black&quot;, trans=&quot;sqrt&quot;, limits=c(0, 0.5)) +
  theme_classic()+
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5),
    strip.text.y = element_text(angle = 0)
  )
p1</code></pre>
<p><img src="figure/results-question-1.Rmd/EACB-heatmap-tcga-rna-genus-1.png" width="300%" height="300%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;data/heatmap_tcgarna_genus.pdf&quot;, p1, units=&quot;in&quot;, height=23, width=16)</code></pre>
<pre class="r"><code>plot.dat &lt;- dat.rna %&gt;% filter(EACcomp != &quot;0&quot;) %&gt;%
  select(EACcomp, Phylum, Genus, Patient_ID, Abundance)
plot.dat &lt;- na.omit(plot.dat) 
  
p1 &lt;- ggplot(plot.dat, aes(x = Patient_ID, y = Phylum, fill = Abundance)) +
  geom_tile()+
  labs(title=&quot;TCGA RNAseq Data&quot;, subtitle = &quot;Phylum Level Data&quot;,
       x = &quot;ID/Accession Number&quot;) +
  facet_grid(.~EACcomp, scales=&quot;free&quot;)+
  scale_fill_gradient(low=&quot;white&quot;, high=&quot;black&quot;, trans=&quot;sqrt&quot;, limits=c(0, 0.001)) +
  theme_classic()+
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5),
    strip.text.y = element_text(angle = 0)
  )
p1</code></pre>
<p><img src="figure/results-question-1.Rmd/EACB-heatmap-tcga-rna-phylum-1.png" width="400%" height="1000%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.dat &lt;- dat.wgs %&gt;% filter(EACcomp != &quot;0&quot;, Abundance &gt; 0.001) %&gt;%
  select(EACcomp, Phylum, Genus, Patient_ID, Abundance)
plot.dat &lt;- na.omit(plot.dat) 

p1 &lt;- ggplot(plot.dat, aes(x = Patient_ID, y = Genus, fill = Abundance)) +
  geom_tile()+
  labs(title=&quot;TCGA WGS Data&quot;, subtitle = &quot;Genus Level Data&quot;,
       x = &quot;ID/Accession Number&quot;) +
  facet_grid(.~EACcomp, scales=&quot;free&quot;)+
  scale_fill_gradient(low=&quot;white&quot;, high=&quot;black&quot;, trans=&quot;sqrt&quot;, limits=c(0, 0.5)) +
  theme_classic()+
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5),
    strip.text.y = element_text(angle = 0)
  )
p1</code></pre>
<p><img src="figure/results-question-1.Rmd/EACB-heatmap-tcga-wgs-genus-1.png" width="300%" height="300%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.dat &lt;- dat.wgs %&gt;% filter(EACcomp != &quot;0&quot;) %&gt;%
  select(EACcomp, Phylum, Genus, Patient_ID, Abundance)
plot.dat &lt;- na.omit(plot.dat) 
  
p1 &lt;- ggplot(plot.dat, aes(x = Patient_ID, y = Phylum, fill = Abundance)) +
  geom_tile()+
  labs(title=&quot;TCGA WGS Data&quot;, subtitle = &quot;Phylum Level Data&quot;,
       x = &quot;ID/Accession Number&quot;) +
  facet_grid(.~EACcomp, scales=&quot;free&quot;)+
  scale_fill_gradient(low=&quot;white&quot;, high=&quot;black&quot;, trans=&quot;sqrt&quot;, limits=c(0, 0.001)) +
  theme_classic()+
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5),
    strip.text.y = element_text(angle = 0)
  )
p1</code></pre>
<p><img src="figure/results-question-1.Rmd/EACB-heatmap-tcga-wgs-phylum-1.png" width="400%" height="1000%" style="display: block; margin: auto;" /></p>
</div>
<div id="specific-otus" class="section level3">
<h3>Specific OTUs</h3>
<pre class="r"><code>plot.dat &lt;- dat.rna.s %&gt;% filter(EACcomp != &quot;0&quot;, is.na(OTU1) == F) %&gt;%
  select(EACcomp, OTU1, Patient_ID, Abundance)
plot.dat &lt;- na.omit(plot.dat) 
  
p1 &lt;- ggplot(plot.dat, aes(x = Patient_ID, y = OTU1, fill = Abundance)) +
  geom_tile()+
  labs(title=&quot;TCGA RNAseq Data&quot;, y=NULL,
       x = &quot;ID/Accession Number&quot;) +
  facet_grid(.~EACcomp, scales=&quot;free&quot;)+
  scale_fill_gradient(low=&quot;white&quot;, high=&quot;black&quot;, trans=&quot;sqrt&quot;, limits=c(0, 0.1)) +
  theme_classic()+
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5),
    strip.text.y = element_text(angle = 0)
  )
p1</code></pre>
<p><img src="figure/results-question-1.Rmd/EACB-heatmap-tcgarna-1.png" width="300%" height="300%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.dat &lt;- dat.wgs.s %&gt;% filter(EACcomp != &quot;0&quot;, is.na(OTU1) == F) %&gt;%
  select(EACcomp, OTU1, Patient_ID, Abundance)
plot.dat &lt;- na.omit(plot.dat)  


p1 &lt;- ggplot(plot.dat, aes(x =Patient_ID, y = OTU1, fill = Abundance)) +
  geom_tile()+
  labs(title=&quot;TCGA WGS Data&quot;, y=NULL,
       x = &quot;ID/Accession Number&quot;) +
  facet_grid(.~EACcomp, scales=&quot;free&quot;)+
  scale_fill_gradient(low=&quot;white&quot;, high=&quot;black&quot;, trans=&quot;sqrt&quot;, limits=c(0, 0.5)) +
  theme_classic()+
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1, vjust=0.5),
    strip.text.y = element_text(angle = 0)
  )
p1</code></pre>
<p><img src="figure/results-question-1.Rmd/EACB-heatmap-tcgawgs-1.png" width="400%" height="100%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="stacked-bar-charts" class="section level2">
<h2>Stacked Bar Charts</h2>
<div id="all-otus-ra-0.001-1" class="section level3">
<h3>All OTUs (RA &gt; 0.001)</h3>
<pre class="r"><code>plot.dat &lt;- dat.rna %&gt;% filter(EACcomp != &quot;0&quot;, Abundance &gt; 0.001)%&gt;%
  dplyr::group_by(EACcomp, Genus)%&gt;%
  dplyr::summarise(
    Abundance = mean(Abundance, na.rm=T)
  )</code></pre>
<pre><code>`summarise()` has grouped output by &#39;EACcomp&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>p1 &lt;- ggplot(plot.dat, aes(x=EACcomp, y = Abundance, fill=Genus)) +
  geom_bar(stat=&quot;identity&quot;)+
  labs(title=&quot;TCGA RNAseq Data&quot;,
       subtitle = &quot;Genus Level Data&quot;,
       x = &quot;Tissue Group&quot;,
       y=&quot;Average Relative Abundance&quot;) +
  theme_classic()
p1</code></pre>
<p><img src="figure/results-question-1.Rmd/EACB-bar-tcga-rna-genus-1.png" width="300%" height="300%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;data/bar_tcgarna_genus.pdf&quot;, p1, units=&quot;in&quot;, height=23, width=16)</code></pre>
<pre class="r"><code>plot.dat &lt;- dat.rna %&gt;% filter(EACcomp != &quot;0&quot;)%&gt;%
  dplyr::group_by(EACcomp, Phylum)%&gt;%
  dplyr::summarise(
    Abundance = mean(Abundance, na.rm=T)
  )</code></pre>
<pre><code>`summarise()` has grouped output by &#39;EACcomp&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>p1 &lt;- ggplot(plot.dat, aes(x=EACcomp, y = Abundance, fill=Phylum)) +
  geom_bar(stat=&quot;identity&quot;)+
  labs(title=&quot;TCGA RNAseq Data&quot;,
       subtitle = &quot;Phylum Level Data&quot;,
       x = &quot;Tissue Group&quot;,
       y=&quot;Average Relative Abundance&quot;) +
  theme_classic()
p1</code></pre>
<p><img src="figure/results-question-1.Rmd/EACB-bar-tcga-rna-phylum-1.png" width="400%" height="1000%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.dat &lt;- dat.wgs %&gt;% filter(EACcomp != &quot;0&quot;, Abundance &gt; 0.001)%&gt;%
  dplyr::group_by(EACcomp, Genus)%&gt;%
  dplyr::summarise(
    Abundance = mean(Abundance, na.rm=T)
  )</code></pre>
<pre><code>`summarise()` has grouped output by &#39;EACcomp&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>p1 &lt;- ggplot(plot.dat, aes(x=EACcomp, y = Abundance, fill=Genus)) +
  geom_bar(stat=&quot;identity&quot;)+
  labs(title=&quot;TCGA WGS Data&quot;,
       subtitle = &quot;Genus Level Data&quot;,
       x = &quot;Tissue Group&quot;,
       y=&quot;Average Relative Abundance&quot;) +
  theme_classic()
p1</code></pre>
<p><img src="figure/results-question-1.Rmd/EACB-bar-tcga-wgs-genus-1.png" width="300%" height="300%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.dat &lt;- dat.wgs %&gt;% filter(EACcomp != &quot;0&quot;)%&gt;%
  dplyr::group_by(EACcomp, Phylum)%&gt;%
  dplyr::summarise(
    Abundance = mean(Abundance, na.rm=T)
  )</code></pre>
<pre><code>`summarise()` has grouped output by &#39;EACcomp&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>p1 &lt;- ggplot(plot.dat, aes(x=EACcomp, y = Abundance, fill=Phylum)) +
  geom_bar(stat=&quot;identity&quot;)+
  labs(title=&quot;TCGA WGS Data&quot;,
       subtitle = &quot;Phylum Level Data&quot;,
       x = &quot;Tissue Group&quot;,
       y=&quot;Average Relative Abundance&quot;) +
  theme_classic()
p1</code></pre>
<p><img src="figure/results-question-1.Rmd/EACB-bar-tcga-wgs-phylum-1.png" width="400%" height="1000%" style="display: block; margin: auto;" /></p>
</div>
<div id="specific-otus-1" class="section level3">
<h3>Specific OTUs</h3>
<pre class="r"><code>plot.dat &lt;- dat.rna.s %&gt;% filter(EACcomp != &quot;0&quot;, is.na(OTU1) == F)%&gt;%
  dplyr::group_by(EACcomp, OTU1)%&gt;%
  dplyr::summarise(
    Abundance = mean(Abundance, na.rm=T)
  )</code></pre>
<pre><code>`summarise()` has grouped output by &#39;EACcomp&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>p1 &lt;- ggplot(plot.dat, aes(x=EACcomp, y = Abundance, fill=OTU1)) +
  geom_bar(stat=&quot;identity&quot;)+
  labs(title=&quot;TCGA RNA seq Data&quot;,
       x = &quot;Tissue Group&quot;,
       y=&quot;Average Relative Abundance&quot;) +
  theme_classic()
p1</code></pre>
<p><img src="figure/results-question-1.Rmd/EACB-bar-tcgarna-1.png" width="300%" height="300%" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot.dat &lt;- dat.wgs.s %&gt;% filter(EACcomp != &quot;0&quot;, is.na(OTU1) == F)%&gt;%
  dplyr::group_by(EACcomp, OTU1)%&gt;%
  dplyr::summarise(
    Abundance = mean(Abundance, na.rm=T)
  )</code></pre>
<pre><code>`summarise()` has grouped output by &#39;EACcomp&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>p1 &lt;- ggplot(plot.dat, aes(x=EACcomp, y = Abundance, fill=OTU1)) +
  geom_bar(stat=&quot;identity&quot;)+
  labs(title=&quot;TCGA WGS Data&quot;,
       x = &quot;Tissue Group&quot;,
       y=&quot;Average Relative Abundance&quot;) +
  theme_classic()
p1</code></pre>
<p><img src="figure/results-question-1.Rmd/EACB-bar-tcgawgs-1.png" width="400%" height="100%" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.3 (2020-10-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] cowplot_1.1.1     dendextend_1.14.0 ggdendro_0.1.22   reshape2_1.4.4   
 [5] car_3.0-10        carData_3.0-4     gvlma_1.0.0.3     patchwork_1.1.1  
 [9] viridis_0.5.1     viridisLite_0.3.0 gridExtra_2.3     xtable_1.8-4     
[13] kableExtra_1.3.1  data.table_1.13.6 readxl_1.3.1      forcats_0.5.0    
[17] stringr_1.4.0     dplyr_1.0.3       purrr_0.3.4       readr_1.4.0      
[21] tidyr_1.1.2       tibble_3.0.5      ggplot2_3.3.3     tidyverse_1.3.0  
[25] lmerTest_3.1-3    lme4_1.1-26       Matrix_1.2-18     vegan_2.5-7      
[29] lattice_0.20-41   permute_0.9-5     phyloseq_1.34.0   workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] minqa_1.2.4         colorspace_2.0-0    rio_0.5.16         
 [4] ellipsis_0.3.1      rprojroot_2.0.2     XVector_0.30.0     
 [7] fs_1.5.0            rstudioapi_0.13     farver_2.0.3       
[10] fansi_0.4.2         lubridate_1.7.9.2   xml2_1.3.2         
[13] codetools_0.2-18    splines_4.0.3       knitr_1.30         
[16] ade4_1.7-16         jsonlite_1.7.2      nloptr_1.2.2.2     
[19] broom_0.7.3         cluster_2.1.0       dbplyr_2.0.0       
[22] BiocManager_1.30.10 compiler_4.0.3      httr_1.4.2         
[25] backports_1.2.0     assertthat_0.2.1    cli_2.2.0          
[28] later_1.1.0.1       htmltools_0.5.1     prettyunits_1.1.1  
[31] tools_4.0.3         igraph_1.2.6        gtable_0.3.0       
[34] glue_1.4.2          Rcpp_1.0.6          Biobase_2.50.0     
[37] cellranger_1.1.0    vctrs_0.3.6         Biostrings_2.58.0  
[40] rhdf5filters_1.2.0  multtest_2.46.0     ape_5.4-1          
[43] nlme_3.1-151        iterators_1.0.13    xfun_0.20          
[46] ps_1.5.0            openxlsx_4.2.3      rvest_0.3.6        
[49] lifecycle_0.2.0     statmod_1.4.35      zlibbioc_1.36.0    
[52] MASS_7.3-53         scales_1.1.1        hms_1.0.0          
[55] promises_1.1.1      parallel_4.0.3      biomformat_1.18.0  
[58] rhdf5_2.34.0        curl_4.3            yaml_2.2.1         
[61] stringi_1.5.3       S4Vectors_0.28.1    foreach_1.5.1      
[64] BiocGenerics_0.36.0 zip_2.1.1           boot_1.3-25        
[67] rlang_0.4.10        pkgconfig_2.0.3     evaluate_0.14      
[70] Rhdf5lib_1.12.0     labeling_0.4.2      tidyselect_1.1.0   
[73] plyr_1.8.6          magrittr_2.0.1      R6_2.5.0           
[76] IRanges_2.24.1      generics_0.1.0      DBI_1.1.1          
[79] foreign_0.8-81      pillar_1.4.7        haven_2.3.1        
[82] whisker_0.4         withr_2.4.0         mgcv_1.8-33        
[85] abind_1.4-5         survival_3.2-7      modelr_0.1.8       
[88] crayon_1.3.4        rmarkdown_2.6       progress_1.2.2     
[91] grid_4.0.3          git2r_0.28.0        reprex_0.3.0       
[94] digest_0.6.27       webshot_0.5.2       httpuv_1.5.5       
[97] numDeriv_2016.8-1.1 stats4_4.0.3        munsell_0.5.0      </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
