<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Sample IDs and Relative Abundance Misc.</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">esoph-micro-cancer</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Sample IDs and Relative Abundance Misc.</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-12-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>esoph-micro-cancer-workflow/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200916code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200916)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200916code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200916)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomGreathouseLabesophmicrocancerworkflowtreecf91029edb7253f0b4c6ea60d979ff38f446b111targetblankcf91029a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/tree/cf91029edb7253f0b4c6ea60d979ff38f446b111" target="_blank">cf91029</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomGreathouseLabesophmicrocancerworkflowtreecf91029edb7253f0b4c6ea60d979ff38f446b111targetblankcf91029a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/tree/cf91029edb7253f0b4c6ea60d979ff38f446b111" target="_blank">cf91029</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/

Untracked files:
    Untracked:  analysis/waterfall-plot.Rmd
    Untracked:  waterfall-plots.Rmd

Unstaged changes:
    Modified:   analysis/data_processing_tcga.Rmd
    Modified:   analysis/test-of-replication.Rmd
    Modified:   code/get_cleaned_data.R
    Modified:   code/get_data.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/test-of-replication.Rmd</code>) and HTML (<code>docs/test-of-replication.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/GreathouseLab/esoph-micro-cancer-workflow/blob/cf91029edb7253f0b4c6ea60d979ff38f446b111/analysis/test-of-replication.Rmd" target="_blank">cf91029</a>
</td>
<td>
noah-padgett
</td>
<td>
2020-12-02
</td>
<td>
updated analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/GreathouseLab/esoph-micro-cancer-workflow/cf91029edb7253f0b4c6ea60d979ff38f446b111/docs/test-of-replication.html" target="_blank">cf91029</a>
</td>
<td>
noah-padgett
</td>
<td>
2020-12-02
</td>
<td>
updated analyses
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<ul>
<li>Goal is to replicate Fisher’s exact test.</li>
</ul>
<div id="data-mung" class="section level1">
<h1>Data mung</h1>
<p>First, we need to format the data for the analyses.</p>
<pre class="r"><code># transform to relative abundances
phylo.data.nci.umd &lt;- transform_sample_counts(phylo.data.nci.umd, function(x){x / sum(x)})
phylo.data.tcga.RNAseq &lt;- transform_sample_counts(phylo.data.tcga.RNAseq, function(x){x / sum(x)})
phylo.data.tcga.WGS &lt;- transform_sample_counts(phylo.data.tcga.WGS, function(x){x / sum(x)})

# melt data down for use
dat.16s &lt;- psmelt(phylo.data.nci.umd)
dat.rna &lt;- psmelt(phylo.data.tcga.RNAseq)
dat.wgs &lt;- psmelt(phylo.data.tcga.WGS)

# fix otu formatting
dat.rna$otu2 &lt;- &quot;a&quot;
dat.wgs$otu2 &lt;- &quot;a&quot;
i &lt;- 1
for(i in 1:nrow(dat.rna)){
  dat.rna$otu2[i] &lt;- str_split(dat.rna$OTU[i], &quot;;&quot;)[[1]][7]
}
for(i in 1:nrow(dat.wgs)){
  dat.wgs$otu2[i] &lt;- str_split(dat.wgs$OTU[i], &quot;;&quot;)[[1]][7]
}


# subset to fuso. nuc. only
# Streptococcus sanguinis 
# Campylobacter concisus
# Prevotella spp.

dat.16s &lt;- filter(
  dat.16s,
  OTU %in% c(
    &quot;Fusobacterium_nucleatum&quot;,
    unique(dat.16s$OTU[dat.16s$OTU %like% &quot;Streptococcus_&quot;]),
    unique(dat.16s$OTU[dat.16s$OTU %like% &quot;Campylobacter_&quot;]),
    &quot;Prevotella_melaninogenica&quot;)
)
dat.rna &lt;- filter(
  dat.rna,
  otu2 %in% c(
    &quot;Fusobacterium nucleatum&quot;,
    &quot;Streptococcus sanguinis&quot;,
      &quot;Streptococcus oralis&quot;,
      &quot;Streptococcus mitis&quot;, 
      &quot;Streptococcus pneumoniae&quot;, 
      &quot;Streptococcus parasanguinis&quot;, 
      &quot;Streptococcus salivarius&quot;,
    &quot;Campylobacter concisus&quot;,
    &quot;Prevotella melaninogenica&quot;)
)
dat.wgs &lt;- filter(
  dat.wgs,
  otu2 %in% c(
    &quot;Fusobacterium nucleatum&quot;,
    &quot;Streptococcus sanguinis&quot;,
      &quot;Streptococcus oralis&quot;,
      &quot;Streptococcus mitis&quot;, 
      &quot;Streptococcus pneumoniae&quot;, 
      &quot;Streptococcus parasanguinis&quot;, 
      &quot;Streptococcus salivarius&quot;,
    &quot;Campylobacter concisus&quot;,
    &quot;Prevotella melaninogenica&quot;)
)

# new names
dat.16s$OTU1 &lt;- factor(
  dat.16s$OTU,
  levels = c(
    &quot;Fusobacterium_nucleatum&quot;,
    &quot;Streptococcus_dentisani:Streptococcus_infantis:Streptococcus_mitis:Streptococcus_oligofermentans:Streptococcus_oralis:Streptococcus_pneumoniae:Streptococcus_pseudopneumoniae:Streptococcus_sanguinis&quot;,
    &quot;Campylobacter_rectus:Campylobacter_showae&quot;,
    &quot;Prevotella_melaninogenica&quot;
  ),
  labels = c(
    &quot;Fusobacterium nucleatum&quot;,
    &quot;Streptococcus sanguinis&quot;,
    &quot;Campylobacter concisus&quot;,
    &quot;Prevotella melaninogenica&quot;
  )
)
dat.rna$OTU1 &lt;- factor(
  dat.rna$otu2,
  levels = c(
    &quot;Fusobacterium nucleatum&quot;,
    &quot;Streptococcus sanguinis&quot;,
    &quot;Campylobacter concisus&quot;,
    &quot;Prevotella melaninogenica&quot;),
  labels = c(
    &quot;Fusobacterium nucleatum&quot;,
    &quot;Streptococcus sanguinis&quot;,
    &quot;Campylobacter concisus&quot;,
    &quot;Prevotella melaninogenica&quot;)
)

dat.wgs$OTU1 &lt;- factor(
  dat.wgs$otu2,
  levels = c(
    &quot;Fusobacterium nucleatum&quot;,
    &quot;Streptococcus sanguinis&quot;,
    &quot;Campylobacter concisus&quot;,
    &quot;Prevotella melaninogenica&quot;),
  labels = c(
    &quot;Fusobacterium nucleatum&quot;,
    &quot;Streptococcus sanguinis&quot;,
    &quot;Campylobacter concisus&quot;,
    &quot;Prevotella melaninogenica&quot;)
)

# rename bacteria
dat.16s$OTU &lt;- factor(
  dat.16s$OTU,
  levels = c(
    &quot;Fusobacterium_nucleatum&quot;,
    &quot;Streptococcus_dentisani:Streptococcus_infantis:Streptococcus_mitis:Streptococcus_oligofermentans:Streptococcus_oralis:Streptococcus_pneumoniae:Streptococcus_pseudopneumoniae:Streptococcus_sanguinis&quot;,
    &quot;Campylobacter_rectus:Campylobacter_showae&quot;,
    &quot;Prevotella_melaninogenica&quot;
  ),
  labels = c(
    &quot;Fusobacterium nucleatum&quot;,
    &quot;Streptococcus spp.&quot;,
    &quot;Campylobacter concisus&quot;,
    &quot;Prevotella melaninogenica&quot;
  )
)
dat.rna$OTU &lt;- factor(
  dat.rna$otu2,
  levels = c(
    &quot;Fusobacterium nucleatum&quot;,
    &quot;Streptococcus sanguinis&quot;,
      &quot;Streptococcus oralis&quot;,
      &quot;Streptococcus mitis&quot;, 
      &quot;Streptococcus pneumoniae&quot;, 
      &quot;Streptococcus parasanguinis&quot;, 
      &quot;Streptococcus salivarius&quot;,
    &quot;Campylobacter concisus&quot;,
    &quot;Prevotella melaninogenica&quot;),
  labels = c(
    &quot;Fusobacterium nucleatum&quot;,
    &quot;Streptococcus spp.&quot;,
      &quot;Streptococcus spp.&quot;,
      &quot;Streptococcus spp.&quot;, 
      &quot;Streptococcus spp.&quot;, 
      &quot;Streptococcus spp.&quot;, 
      &quot;Streptococcus spp.&quot;,
    &quot;Campylobacter concisus&quot;,
    &quot;Prevotella melaninogenica&quot;)
)

dat.wgs$OTU &lt;- factor(
  dat.wgs$otu2,
  levels = c(
    &quot;Fusobacterium nucleatum&quot;,
    &quot;Streptococcus sanguinis&quot;,
      &quot;Streptococcus oralis&quot;,
      &quot;Streptococcus mitis&quot;, 
      &quot;Streptococcus pneumoniae&quot;, 
      &quot;Streptococcus parasanguinis&quot;, 
      &quot;Streptococcus salivarius&quot;,
    &quot;Campylobacter concisus&quot;,
    &quot;Prevotella melaninogenica&quot;),
  labels = c(
    &quot;Fusobacterium nucleatum&quot;,
    &quot;Streptococcus spp.&quot;,
      &quot;Streptococcus spp.&quot;,
      &quot;Streptococcus spp.&quot;, 
      &quot;Streptococcus spp.&quot;, 
      &quot;Streptococcus spp.&quot;, 
      &quot;Streptococcus spp.&quot;,
    &quot;Campylobacter concisus&quot;,
    &quot;Prevotella melaninogenica&quot;)
)



# make tumor vs normal variable
dat.16s$tumor.cat &lt;- factor(dat.16s$tissue, levels=c(&quot;BO&quot;, &quot;N&quot;, &quot;T&quot;), labels = c(&quot;Non-Tumor&quot;, &quot;Non-Tumor&quot;, &quot;Tumor&quot;))
dat.rna$tumor.cat &lt;- factor(dat.rna$SampleType_Level2, levels=c(&quot;Normal&quot;, &quot;Tumor&quot;), labels = c(&quot;Non-Tumor&quot;, &quot;Tumor&quot;))
dat.wgs$tumor.cat &lt;- factor(dat.wgs$SampleType_Level2, levels=c(&quot;Normal&quot;, &quot;Tumor&quot;), labels = c(&quot;Non-Tumor&quot;, &quot;Tumor&quot;))

# dataset id
dat.16s$source &lt;- &quot;16s&quot;
dat.rna$source &lt;- &quot;rna&quot;
dat.wgs$source &lt;- &quot;wgs&quot;

# plotting ids
dat.16s$X &lt;- paste0(dat.16s$source, &quot;-&quot;, dat.16s$tumor.cat)
dat.rna$X &lt;- paste0(dat.rna$source, &quot;-&quot;, dat.rna$tumor.cat)
dat.wgs$X &lt;- paste0(dat.wgs$source, &quot;-&quot;, dat.wgs$tumor.cat)

# relabel as (0/1) for analysis
dat.16s$tumor &lt;- as.numeric(factor(dat.16s$tissue, levels=c(&quot;BO&quot;, &quot;N&quot;, &quot;T&quot;), labels = c(&quot;Non-Tumor&quot;, &quot;Non-Tumor&quot;, &quot;Tumor&quot;))) - 1
dat.rna$tumor &lt;- as.numeric(factor(dat.rna$SampleType_Level2, levels=c(&quot;Normal&quot;, &quot;Tumor&quot;), labels = c(&quot;Non-Tumor&quot;, &quot;Tumor&quot;))) - 1
dat.wgs$tumor &lt;- as.numeric(factor(dat.wgs$SampleType_Level2, levels=c(&quot;Normal&quot;, &quot;Tumor&quot;), labels = c(&quot;Non-Tumor&quot;, &quot;Tumor&quot;))) - 1

# presence- absence
dat.16s$pres &lt;- ifelse(dat.16s$Abundance &gt; 0, 1, 0)
dat.16s$pres[is.na(dat.16s$pres)] &lt;- 0
dat.rna$pres &lt;- ifelse(dat.rna$Abundance &gt; 0, 1, 0)
dat.rna$pres[is.na(dat.rna$pres)] &lt;- 0
dat.wgs$pres &lt;- ifelse(dat.wgs$Abundance &gt; 0, 1, 0)
dat.wgs$pres[is.na(dat.wgs$pres)] &lt;- 0

# blood exclusions
dat.16s$bloodWGS &lt;- 0
dat.rna$bloodWGS &lt;- dat.rna$WGS_Blood_Normal
dat.wgs$bloodWGS &lt;- dat.wgs$WGS_Blood_Normal

# merge data
cls &lt;- c(&quot;OTU&quot;, &quot;OTU1&quot;, &quot;Sample&quot;, &quot;Abundance&quot;, &quot;tumor.cat&quot;, &quot;tumor&quot;, &quot;source&quot;, &quot;X&quot;, &quot;pres&quot;, &quot;bloodWGS&quot;)
mydata &lt;- full_join(dat.16s[,cls], dat.rna[,cls])</code></pre>
<pre><code>Joining, by = c(&quot;OTU&quot;, &quot;OTU1&quot;, &quot;Sample&quot;, &quot;Abundance&quot;, &quot;tumor.cat&quot;, &quot;tumor&quot;, &quot;source&quot;, &quot;X&quot;, &quot;pres&quot;, &quot;bloodWGS&quot;)</code></pre>
<pre class="r"><code>mydata &lt;- full_join(mydata, dat.wgs[,cls])</code></pre>
<pre><code>Joining, by = c(&quot;OTU&quot;, &quot;OTU1&quot;, &quot;Sample&quot;, &quot;Abundance&quot;, &quot;tumor.cat&quot;, &quot;tumor&quot;, &quot;source&quot;, &quot;X&quot;, &quot;pres&quot;, &quot;bloodWGS&quot;)</code></pre>
</div>
<div id="replicating-the-analysis" class="section level1">
<h1>Replicating the Analysis</h1>
<div id="plot" class="section level2">
<h2>Plot</h2>
<pre class="r"><code>p &lt;- ggplot(mydata, aes(x=X, y=Abundance)) +
  geom_violin() +
  geom_jitter(alpha=0.5) +
  facet_wrap(.~OTU)+
  scale_y_continuous(
    trans = &quot;sqrt&quot;,
    breaks=c(0.002, 0.01, 0.05, 0.1, 0.25, 0.5, 0.8)) +
  labs(x=NULL, y=&quot;Relative Abundance&quot;)+
  theme_classic() +
  theme(
    axis.text.x = element_text(angle = 30,vjust=0.95, hjust=0.95)
  )
p</code></pre>
<pre><code>Warning: Removed 1107 rows containing non-finite values (stat_ydensity).</code></pre>
<pre><code>Warning: Removed 1107 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/test-of-replication.Rmd/unnamed-chunk-2-1.png" width="100%" height="200%" style="display: block; margin: auto;" /></p>
</div>
<div id="statistical-tests" class="section level2">
<h2>Statistical Tests</h2>
<div id="fusobacterium-nucleatum" class="section level3">
<h3>Fusobacterium nucleatum</h3>
<pre class="r"><code># No additional subsetting/exclusions
d &lt;- mydata %&gt;%
  filter(OTU == &quot;Fusobacterium nucleatum&quot;,
         source == &quot;16s&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 0.1487
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.8126368 3.2144776
sample estimates:
odds ratio 
  1.610386 </code></pre>
<pre class="r"><code>d &lt;- mydata %&gt;%
  filter(OTU == &quot;Fusobacterium nucleatum&quot;,
         source == &quot;wgs&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 0.02533
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 1.065529 4.759083
sample estimates:
odds ratio 
  2.232631 </code></pre>
<pre class="r"><code>d &lt;- mydata %&gt;%
  filter(OTU == &quot;Fusobacterium nucleatum&quot;,
         source == &quot;rna&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 0.04707
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.05593801 1.12690374
sample estimates:
odds ratio 
 0.2723903 </code></pre>
<pre class="r"><code># Excluding blood != 0
d &lt;- mydata %&gt;%
  filter(OTU == &quot;Fusobacterium nucleatum&quot;,
         source == &quot;wgs&quot;,
         bloodWGS==0)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 0.7626
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.3753539 5.6058176
sample estimates:
odds ratio 
  1.432327 </code></pre>
</div>
<div id="streptococcus-spp." class="section level3">
<h3>Streptococcus spp.</h3>
<p><em>difficult to determine which species were used</em></p>
<pre class="r"><code># No additional subsetting/exclusions
d &lt;- mydata %&gt;%
  filter(OTU == &quot;Streptococcus spp.&quot;,
         source == &quot;16s&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.2189448 7.8326544
sample estimates:
odds ratio 
  1.173049 </code></pre>
<pre class="r"><code>d &lt;- mydata %&gt;%
  filter(OTU == &quot;Streptococcus spp.&quot;,
         source == &quot;wgs&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 1.413e-06
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 1.487867 2.643073
sample estimates:
odds ratio 
  1.981098 </code></pre>
<pre class="r"><code>d &lt;- mydata %&gt;%
  filter(OTU == &quot;Streptococcus spp.&quot;,
         source == &quot;rna&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 8.651e-07
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.1589282 0.4818080
sample estimates:
odds ratio 
 0.2795264 </code></pre>
<pre class="r"><code># Excluding blood != 0
d &lt;- mydata %&gt;%
  filter(OTU == &quot;Streptococcus spp.&quot;,
         source == &quot;wgs&quot;,
         bloodWGS==0)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.5998557 1.6670677
sample estimates:
odds ratio 
         1 </code></pre>
</div>
<div id="streptococcus-sanguinis" class="section level3">
<h3>Streptococcus sanguinis</h3>
<p><em>difficult to determine which species were used</em></p>
<pre class="r"><code># No additional subsetting/exclusions
d &lt;- mydata %&gt;%
  filter(OTU1 == &quot;Streptococcus sanguinis&quot;,
         source == &quot;16s&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.2189448 7.8326544
sample estimates:
odds ratio 
  1.173049 </code></pre>
<pre class="r"><code>d &lt;- mydata %&gt;%
  filter(OTU1 == &quot;Streptococcus sanguinis&quot;,
         source == &quot;wgs&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 0.4639
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.6098689 2.9683346
sample estimates:
odds ratio 
  1.339246 </code></pre>
<pre class="r"><code>d &lt;- mydata %&gt;%
  filter(OTU1 == &quot;Streptococcus sanguinis&quot;,
         source == &quot;rna&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 0.04205
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.05280152 1.06581611
sample estimates:
odds ratio 
 0.2573433 </code></pre>
<pre class="r"><code># Excluding blood != 0
d &lt;- mydata %&gt;%
  filter(OTU1 == &quot;Streptococcus sanguinis&quot;,
         source == &quot;wgs&quot;,
         bloodWGS==0)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.2622982 6.4033751
sample estimates:
odds ratio 
  1.267969 </code></pre>
</div>
<div id="campylobacter-concisus" class="section level3">
<h3>Campylobacter concisus</h3>
<pre class="r"><code># No additional subsetting/exclusions
d &lt;- mydata %&gt;%
  filter(OTU == &quot;Campylobacter concisus&quot;,
         source == &quot;16s&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 0.2325
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.7306314 3.9724225
sample estimates:
odds ratio 
  1.698461 </code></pre>
<pre class="r"><code>d &lt;- mydata %&gt;%
  filter(OTU == &quot;Campylobacter concisus&quot;,
         source == &quot;wgs&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 0.008873
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 1.225362 6.656564
sample estimates:
odds ratio 
  2.797326 </code></pre>
<pre class="r"><code>d &lt;- mydata %&gt;%
  filter(OTU == &quot;Campylobacter concisus&quot;,
         source == &quot;rna&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 0.04188
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.05364598 1.20067358
sample estimates:
odds ratio 
  0.235494 </code></pre>
<pre class="r"><code># Excluding blood != 0
d &lt;- mydata %&gt;%
  filter(OTU == &quot;Campylobacter concisus&quot;,
         source == &quot;wgs&quot;,
         bloodWGS==0)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 1
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.2461498 4.0625664
sample estimates:
odds ratio 
         1 </code></pre>
</div>
<div id="prevotella-melaninogenica" class="section level3">
<h3>Prevotella melaninogenica</h3>
<pre class="r"><code># No additional subsetting/exclusions
d &lt;- mydata %&gt;%
  filter(OTU == &quot;Prevotella melaninogenica&quot;,
         source == &quot;16s&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 0.1531
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.790056 3.924226
sample estimates:
odds ratio 
  1.728747 </code></pre>
<pre class="r"><code>d &lt;- mydata %&gt;%
  filter(OTU == &quot;Prevotella melaninogenica&quot;,
         source == &quot;wgs&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 3.232e-06
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  2.492672 12.298866
sample estimates:
odds ratio 
  5.430828 </code></pre>
<pre class="r"><code>d &lt;- mydata %&gt;%
  filter(OTU == &quot;Prevotella melaninogenica&quot;,
         source == &quot;rna&quot;)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 0.1884
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.09378767 1.68318325
sample estimates:
odds ratio 
 0.4075071 </code></pre>
<pre class="r"><code># Excluding blood != 0
d &lt;- mydata %&gt;%
  filter(OTU == &quot;Prevotella melaninogenica&quot;,
         source == &quot;wgs&quot;,
         bloodWGS==0)
fisher.test(d$pres, d$tumor)</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  d$pres and d$tumor
p-value = 0.747
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
 0.355727 6.731657
sample estimates:
odds ratio 
  1.509201 </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18363)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] car_3.0-8         carData_3.0-4     gvlma_1.0.0.3     patchwork_1.0.1  
 [5] viridis_0.5.1     viridisLite_0.3.0 gridExtra_2.3     xtable_1.8-4     
 [9] kableExtra_1.1.0  plyr_1.8.6        data.table_1.13.0 readxl_1.3.1     
[13] forcats_0.5.0     stringr_1.4.0     dplyr_1.0.1       purrr_0.3.4      
[17] readr_1.3.1       tidyr_1.1.1       tibble_3.0.3      ggplot2_3.3.2    
[21] tidyverse_1.3.0   lmerTest_3.1-2    lme4_1.1-23       Matrix_1.2-18    
[25] vegan_2.5-6       lattice_0.20-41   permute_0.9-5     phyloseq_1.32.0  
[29] workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] minqa_1.2.4         colorspace_1.4-1    rio_0.5.16         
 [4] ellipsis_0.3.1      rprojroot_1.3-2     XVector_0.28.0     
 [7] fs_1.5.0            rstudioapi_0.11     farver_2.0.3       
[10] fansi_0.4.1         lubridate_1.7.9     xml2_1.3.2         
[13] codetools_0.2-16    splines_4.0.2       knitr_1.29         
[16] ade4_1.7-15         jsonlite_1.7.0      nloptr_1.2.2.2     
[19] broom_0.7.0         cluster_2.1.0       dbplyr_1.4.4       
[22] BiocManager_1.30.10 compiler_4.0.2      httr_1.4.2         
[25] backports_1.1.7     assertthat_0.2.1    cli_2.0.2          
[28] later_1.1.0.1       htmltools_0.5.0     tools_4.0.2        
[31] igraph_1.2.5        gtable_0.3.0        glue_1.4.1         
[34] reshape2_1.4.4      Rcpp_1.0.5          Biobase_2.48.0     
[37] cellranger_1.1.0    vctrs_0.3.2         Biostrings_2.56.0  
[40] multtest_2.44.0     ape_5.4             nlme_3.1-148       
[43] iterators_1.0.12    xfun_0.19           openxlsx_4.1.5     
[46] rvest_0.3.6         lifecycle_0.2.0     statmod_1.4.34     
[49] zlibbioc_1.34.0     MASS_7.3-51.6       scales_1.1.1       
[52] hms_0.5.3           promises_1.1.1      parallel_4.0.2     
[55] biomformat_1.16.0   rhdf5_2.32.2        curl_4.3           
[58] yaml_2.2.1          stringi_1.4.6       S4Vectors_0.26.1   
[61] foreach_1.5.0       BiocGenerics_0.34.0 zip_2.0.4          
[64] boot_1.3-25         rlang_0.4.7         pkgconfig_2.0.3    
[67] evaluate_0.14       Rhdf5lib_1.10.1     tidyselect_1.1.0   
[70] magrittr_1.5        R6_2.4.1            IRanges_2.22.2     
[73] generics_0.0.2      DBI_1.1.0           foreign_0.8-80     
[76] pillar_1.4.6        haven_2.3.1         whisker_0.4        
[79] withr_2.2.0         mgcv_1.8-31         abind_1.4-5        
[82] survival_3.2-3      modelr_0.1.8        crayon_1.3.4       
[85] rmarkdown_2.5       grid_4.0.2          blob_1.2.1         
[88] git2r_0.27.1        reprex_0.3.0        digest_0.6.25      
[91] webshot_0.5.2       httpuv_1.5.4        numDeriv_2016.8-1.1
[94] stats4_4.0.2        munsell_0.5.0      </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
