---
title: "TCGA: Data Processing, Checking, and Exploration"
output:
  workflowr::wflow_html:
    toc: true
editor_options:
  chunk_output_type: console
---


For the TCGA data, data need to be processed twice. First, for the RNAseq microbiome data. Next, for the WGS microbiome data.
When you try to do both at the same time, then there is a mismatch among cases with respect to the number of samples that were generated for each case.


This page contains the investigation of the raw data (OTUs) to identify if outliers are present or whether other issues emerge that may influence our results in unexpected ways.
This file goes through the following checks:

1. Removal of Phylum NA features
2. Computation of total and average prevalence in each Phylum
3. Removal Phyla with 1% or less of all samples
4. Computation of total read count for each Phyla
5. Plotting taxa prevalence vs total counts - identify a natural threshold if clear, if not use 5%
6. Merging taxa to genus rank/level 
7. Abundance Value Transformations
8. Plotting of abundance values by "SampleType_Level2: Tumor, Normal" before transformation and after



```{r data, include=FALSE, echo=FALSE}
# load packages
source("code/load_packages.R")

# get data
source("code/get_data.R")

theme_set(theme_bw())
pal = "Sequential"
scale_colour_discrete <-  function(palname=pal, ...){
  scale_colour_brewer(palette=palname, ...)
}
scale_fill_discrete <-  function(palname=pal, ...){
  scale_fill_brewer(palette=palname, ...)
}

knitr::opts_chunk$set(out.width = "225%", out.height = 10)
```

# RNAseq Data: Taxonomic Filtering

## 0. Sample Reads, Totals, and Rarifying

```{r 0-rarify }

sampleReads <- phyloseq::sample_sums(phylo.data.tcga.RNAseq)

# Total quality Reads
sum(sampleReads)

# Average reads
mean(sampleReads)

# max sequencing depth
max(sampleReads)

# rarified to an even depth of
phylo.data.tcga <- phyloseq::rarefy_even_depth(phylo.data.tcga.RNAseq, replace = T, rngseed = 20200923)

# even depth of:
phyloseq::sample_sums(phylo.data.tcga)

```

## 1. Removal of Phylum NA features

```{r 1-removal }

# show ranks
phyloseq::rank_names(phylo.data.tcga)

# table of features for each phylum
table(phyloseq::tax_table(phylo.data.tcga)[,"Phylum"], exclude=NULL)

```

Note that no taxa were labels as *NA* so none were removed.

## 2. Computation of total and average prevalence in each Phylum

```{r 2-comp-1 }
# compute prevalence of each feature
prevdf <- apply(X=phyloseq::otu_table(phylo.data.tcga), 
                MARGIN= ifelse(phyloseq::taxa_are_rows(phylo.data.tcga), yes=1, no=2),
                FUN=function(x){sum(x>0)})
# store as data.frame with labels
prevdf <- data.frame(Prevalence=prevdf,
                     TotalAbundance=phyloseq::taxa_sums(phylo.data.tcga),
                     phyloseq::tax_table(phylo.data.tcga))

```

Compute the totals and averages abundances.

```{r 2-comp-2 }

totals <- plyr::ddply(prevdf, "Phylum",
            function(df1){
              A <- cbind(mean(df1$Prevalence), sum(df1$Prevalence))
              colnames(A) <- c("Average", "Total")
              A
              }
            ) # end

totals

```

Any of the taxa under a total of 100 may be suspect.
First, we will remove the taxa that are clearly too low in abundance (<=5).

```{r 2-comp-3 }

filterPhyla <- totals$Phylum[totals$Total <= 5, drop=T] # drop allows some of the attributes to be removed

phylo.data1 <- phyloseq::subset_taxa(phylo.data.tcga, !Phylum %in% filterPhyla)
phylo.data1

```

Next, we explore the taxa in more detail next as we move to remove some of these low abundance taxa.


## 3. Removal Phyla with 1% or less of all samples (prevalence filtering)


```{r 3-remove-phylum-1 }

prevdf1 <- subset(prevdf, Phylum %in% phyloseq::get_taxa_unique(phylo.data1, "Phylum"))


```


## 4. Total count computation

```{r 4-count-tot }

# already done above ()

```


## 5. Threshold identification

```{r 5-thresh-1 }
ggplot(prevdf1, aes(TotalAbundance+1,
                    Prevalence/nsamples(phylo.data.tcga))) +
  geom_hline(yintercept=0.01, alpha=0.5, linetype=2)+
  geom_point(size=2, alpha=0.75) +
  scale_x_log10()+
  labs(x="Total Abundance", y="Prevalance [Frac. Samples]")+
  facet_wrap(.~Phylum) + theme(legend.position = "none")

```

Note: for plotting purposes, a $+1$ was added to all TotalAbundances to avoid a taking the log of 0. 


Next, we define a prevalence threshold, that way the taxa can be pruned to a prespecified level. 
In this study, we used 0.01 (1\%) of total samples.

```{r 3-prev-threshold }

prevalenceThreshold <- 0.01*(phyloseq::nsamples(phylo.data.tcga))
prevalenceThreshold

# execute the filtering to this level
keepTaxa <- rownames(prevdf1)[(prevdf1$Prevalence >= prevalenceThreshold)]
phylo.data2 <- phyloseq::prune_taxa(keepTaxa, phylo.data1)

```


## 6. Merge taxa (to genus level)

```{r 4-merge-taxa }
genusNames <- phyloseq::get_taxa_unique(phylo.data2, "Genus")
#phylo.data3 <- merge_taxa(phylo.data2, genusNames, genusNames[which.max(taxa_sums(phylo.data2)[genusNames])])


# How many genera would be present after filtering?
length(phyloseq::get_taxa_unique(phylo.data2, taxonomic.rank = "Genus"))

## [1] 144

phylo.data3 = phyloseq::tax_glom(phylo.data2, "Genus", NArm = TRUE)

```

## 7. Relative Adbundance Plot

```{r 4-rel-abund }

plot_abundance = function(physeq, title = "", ylab="Abundance"){
  mphyseq = phyloseq::psmelt(physeq)
  mphyseq <- subset(mphyseq, Abundance > 0)
  ggplot(data = mphyseq, aes(x=SampleType_Level2, y=Abundance)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position="none")
}


# Transform to relative abundance. Save as new object.
phylo.data3ra = transform_sample_counts(phylo.data3, function(x){x / sum(x)})


plotBefore = plot_abundance(phylo.data3, ylab="Abundance prior to transformation")
plotAfter = plot_abundance(phylo.data3ra, ylab="Relative Abundance")
# Combine each plot into one graphic.
plotBefore + plotAfter + plot_layout(nrow=2)


```

## 8. Plotting Abundance

### Abundance by Phylum

```{r 4-phylum }

plot_abundance = function(physeq, title = "", Facet = "Phylum",
                          ylab="Abundance"){
  mphyseq = phyloseq::psmelt(physeq)
  mphyseq <- subset(mphyseq, Abundance > 0)
  ggplot(data = mphyseq, aes(x=SampleType_Level2, y=Abundance)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position="none")
}



plotBefore = plot_abundance(phylo.data3, ylab="Abundance prior to transformation")
plotAfter = plot_abundance(phylo.data3ra, ylab="Relative Abundance")
# Combine each plot into one graphic.
plotBefore + plotAfter + plot_layout(nrow=2)

```


#### Phylum: Bacteroidetes

```{r 4-rel-abund-2 }

plot_abundance = function(physeq, title = "",
                          Facet = "Phylum",
                          ylab="Abundance"){
  mphyseq = phyloseq::subset_taxa(physeq, Phylum %in% "Bacteroidetes")
  mphyseq <- phyloseq::psmelt(mphyseq)
  mphyseq <- subset(mphyseq, Abundance > 0)
  ggplot(data = mphyseq, aes(x=SampleType_Level2, y=Abundance)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position="none")
}

plotBefore = plot_abundance(phylo.data3,
                            ylab="Abundance prior to transformation")
plotAfter = plot_abundance(phylo.data3ra,
                           ylab="Relative Abundance")

plotBefore + plotAfter + plot_layout(nrow=2)

```


#### Phylum: Firmicutes

```{r 4-rel-abund-3 }

plot_abundance = function(physeq, title = "", Facet = "Class", ylab="Abundance"){
  mphyseq = phyloseq::subset_taxa(physeq, Phylum %in% "Firmicutes")
  mphyseq <- phyloseq::psmelt(mphyseq)
  mphyseq <- subset(mphyseq, Abundance > 0)
  ggplot(data = mphyseq, aes(x=SampleType_Level2, y=Abundance)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position="none")
}

plotBefore = plot_abundance(phylo.data3,
                            ylab="Abundance prior to transformation")
plotAfter = plot_abundance(phylo.data3ra,
                           ylab="Relative Abundance")

plotBefore + plotAfter + plot_layout(nrow=2)


```


#### Phylum: Fusobacteria

```{r 4-rel-abund-4 }

plot_abundance = function(physeq, title = "", Facet = "Class", ylab="Abundance"){
  mphyseq = phyloseq::subset_taxa(physeq, Phylum %in% "Fusobacteria")
  mphyseq <- phyloseq::psmelt(mphyseq)
  mphyseq <- subset(mphyseq, Abundance > 0)
  ggplot(data = mphyseq, aes(x=SampleType_Level2, y=Abundance)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position="none")
}

plotBefore = plot_abundance(phylo.data3,
                            ylab="Abundance prior to transformation")
plotAfter = plot_abundance(phylo.data3ra,
                           ylab="Relative Abundance")

plotBefore + plotAfter + plot_layout(nrow=2)

```


#### Phylum: Proteobacteria

```{r 4-rel-abund-5 }

plot_abundance = function(physeq, title = "", Facet = "Class", ylab="Abundance"){
  mphyseq = phyloseq::subset_taxa(physeq, Phylum %in% "Proteobacteria")
  mphyseq <- phyloseq::psmelt(mphyseq)
  mphyseq <- subset(mphyseq, Abundance > 0)
  ggplot(data = mphyseq, aes(x=SampleType_Level2, y=Abundance)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position="none")
}

plotBefore = plot_abundance(phylo.data3,
                            ylab="Abundance prior to transformation")
plotAfter = plot_abundance(phylo.data3ra,
                           ylab="Relative Abundance")

plotBefore + plotAfter + plot_layout(nrow=2)

```




# WGS Data: Taxonomic Filtering

## 0. Sample Reads, Totals, and Rarifying

```{r WGS-0-rarify }

sampleReads <- sample_sums(phylo.data.tcga.WGS)

# Total quality Reads
sum(sampleReads)

# Average reads
mean(sampleReads)

# max sequencing depth
max(sampleReads)

# rarified to an even depth of
phylo.data.tcga <- phylo.data.tcga.WGS #rarefy_even_depth(phylo.data.tcga.WGS, replace = T, rngseed = 20200923)

# even depth of:
sample_sums(phylo.data.tcga)

```

## 1. Removal of Phylum NA features

```{r WGS-1-removal }

# show ranks
rank_names(phylo.data.tcga)

# table of features for each phylum
table(tax_table(phylo.data.tcga)[,"Phylum"], exclude=NULL)

```

Note that no taxa were labels as *NA* so none were removed.

## 2. Computation of total and average prevalence in each Phylum

```{r WGS-2-comp-1 }
# compute prevalence of each feature
prevdf <- apply(X=otu_table(phylo.data.tcga), 
                MARGIN= ifelse(taxa_are_rows(phylo.data.tcga), yes=1, no=2),
                FUN=function(x){sum(x>0)})
# store as data.frame with labels
prevdf <- data.frame(Prevalence=prevdf,
                     TotalAbundance=taxa_sums(phylo.data.tcga),
                     tax_table(phylo.data.tcga))

```

Compute the totals and averages abundances.

```{r WGS-2-comp-2 }

totals <- plyr::ddply(prevdf, "Phylum",
            function(df1){
              A <- cbind(mean(df1$Prevalence), sum(df1$Prevalence))
              colnames(A) <- c("Average", "Total")
              A
              }
            ) # end

totals

```

Any of the taxa under a total of 100 may be suspect.
First, we will remove the taxa that are clearly too low in abundance (<=5).

```{r WGS-2-comp-3 }

filterPhyla <- totals$Phylum[totals$Total <= 5, drop=T] # drop allows some of the attributes to be removed

phylo.data1 <- subset_taxa(phylo.data.tcga, !Phylum %in% filterPhyla)
phylo.data1

```

Next, we explore the taxa in more detail next as we move to remove some of these low abundance taxa.


## 3. Removal Phyla with 1% or less of all samples (prevalence filtering)


```{r WGS-3-remove-phylum-1 }

prevdf1 <- subset(prevdf, Phylum %in% get_taxa_unique(phylo.data1, "Phylum"))


```


## 4. Total count computation

```{r WGS-4-count-tot }

# already done above ()

```


## 5. Threshold identification

```{r WGS-5-thresh-1 }
ggplot(prevdf1, aes(TotalAbundance+1,
                    Prevalence/nsamples(phylo.data.tcga))) +
  geom_hline(yintercept=0.01, alpha=0.5, linetype=2)+
  geom_point(size=2, alpha=0.75) +
  scale_x_log10()+
  labs(x="Total Abundance", y="Prevalance [Frac. Samples]")+
  facet_wrap(.~Phylum) + theme(legend.position = "none")

```

Note: for plotting purposes, a $+1$ was added to all TotalAbundances to avoid a taking the log of 0. 


Next, we define a prevalence threshold, that way the taxa can be pruned to a prespecified level. 
In this study, we used 0.01 (1\%) of total samples.

```{r WGS-3-prev-threshold }

prevalenceThreshold <- 0.01*nsamples(phylo.data.tcga)
prevalenceThreshold

# execute the filtering to this level
keepTaxa <- rownames(prevdf1)[(prevdf1$Prevalence >= prevalenceThreshold)]
phylo.data2 <- prune_taxa(keepTaxa, phylo.data1)

```


## 6. Merge taxa (to genus level)

```{r WGS-4-merge-taxa }
genusNames <- get_taxa_unique(phylo.data2, "Genus")
#phylo.data3 <- merge_taxa(phylo.data2, genusNames, genusNames[which.max(taxa_sums(phylo.data2)[genusNames])])


# How many genera would be present after filtering?
length(get_taxa_unique(phylo.data2, taxonomic.rank = "Genus"))

## [1] 371

phylo.data3 = tax_glom(phylo.data2, "Genus", NArm = TRUE)

```

## 7. Relative Adbundance Plot

```{r WGS-4-rel-abund }

plot_abundance = function(physeq, title = "", ylab="Abundance"){
  mphyseq = psmelt(physeq)
  mphyseq <- subset(mphyseq, Abundance > 0)
  ggplot(data = mphyseq, aes(x=SampleType_Level2, y=Abundance)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position="none")
}


# Transform to relative abundance. Save as new object.
phylo.data3ra = transform_sample_counts(phylo.data3, function(x){x / sum(x)})


plotBefore = plot_abundance(phylo.data3, ylab="Abundance prior to transformation")
plotAfter = plot_abundance(phylo.data3ra, ylab="Relative Abundance")
# Combine each plot into one graphic.
plotBefore + plotAfter + plot_layout(nrow=2)


```

## 8. Plotting Abundance

### Abundance by Phylum

```{r WGS-4-phylum }

plot_abundance = function(physeq, title = "", Facet = "Phylum", ylab="Abundance"){
  mphyseq <- phyloseq::psmelt(physeq)
  mphyseq <- subset(mphyseq, Abundance > 0)
  ggplot(data = mphyseq, aes(x=SampleType_Level2, y=Abundance)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position="none")
}

plotBefore = plot_abundance(phylo.data3,
                            ylab="Abundance prior to transformation")
plotAfter = plot_abundance(phylo.data3ra,
                           ylab="Relative Abundance")

plotBefore + plotAfter + plot_layout(nrow=2)
```


#### Phylum: Bacteroidetes

```{r WGS-4-rel-abund-2 }

plot_abundance = function(physeq, title = "", Facet = "Class", ylab="Abundance"){
  mphyseq = phyloseq::subset_taxa(physeq, Phylum %in% "Bacteroidetes")
  mphyseq <- phyloseq::psmelt(mphyseq)
  mphyseq <- subset(mphyseq, Abundance > 0)
  ggplot(data = mphyseq, aes(x=SampleType_Level2, y=Abundance)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position="none")
}

plotBefore = plot_abundance(phylo.data3,
                            ylab="Abundance prior to transformation")
plotAfter = plot_abundance(phylo.data3ra,
                           ylab="Relative Abundance")

plotBefore + plotAfter + plot_layout(nrow=2)

```


#### Phylum: Firmicutes

```{r WGS-4-rel-abund-3 }

plot_abundance = function(physeq, title = "", Facet = "Class", ylab="Abundance"){
  mphyseq = phyloseq::subset_taxa(physeq, Phylum %in% "Firmicutes")
  mphyseq <- phyloseq::psmelt(mphyseq)
  mphyseq <- subset(mphyseq, Abundance > 0)
  ggplot(data = mphyseq, aes(x=SampleType_Level2, y=Abundance)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position="none")
}

plotBefore = plot_abundance(phylo.data3,
                            ylab="Abundance prior to transformation")
plotAfter = plot_abundance(phylo.data3ra,
                           ylab="Relative Abundance")

plotBefore + plotAfter + plot_layout(nrow=2)
```


#### Phylum: Fusobacteria

```{r WGS-4-rel-abund-4 }

plot_abundance = function(physeq, title = "", Facet = "Class", ylab="Abundance"){
  mphyseq = phyloseq::subset_taxa(physeq, Phylum %in% "Fusobacteria")
  mphyseq <- phyloseq::psmelt(mphyseq)
  mphyseq <- subset(mphyseq, Abundance > 0)
  ggplot(data = mphyseq, aes(x=SampleType_Level2, y=Abundance)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position="none")
}

plotBefore = plot_abundance(phylo.data3,
                            ylab="Abundance prior to transformation")
plotAfter = plot_abundance(phylo.data3ra,
                           ylab="Relative Abundance")

plotBefore + plotAfter + plot_layout(nrow=2)


```


#### Phylum: Proteobacteria

```{r WGS-4-rel-abund-5 }

plot_abundance = function(physeq, title = "", Facet = "Class", ylab="Abundance"){
  mphyseq = phyloseq::subset_taxa(physeq, Phylum %in% "Proteobacteria")
  mphyseq <- phyloseq::psmelt(mphyseq)
  mphyseq <- subset(mphyseq, Abundance > 0)
  ggplot(data = mphyseq, aes(x=SampleType_Level2, y=Abundance)) +
    geom_violin(fill = NA) +
    geom_point(size = 1, alpha = 0.9,
    position = position_jitter(width = 0.3)) +
    facet_wrap(facets = Facet) + scale_y_log10()+
    labs(y=ylab)+
    theme(legend.position="none")
}

plotBefore = plot_abundance(phylo.data3,
                            ylab="Abundance prior to transformation")
plotAfter = plot_abundance(phylo.data3ra,
                           ylab="Relative Abundance")

plotBefore + plotAfter + plot_layout(nrow=2)

```
